
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../Experiment-Database/">
      
      
        <link rel="next" href="../music/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Genomic Resources - Roberts Lab Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#acropora-pulchra" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Roberts Lab Handbook" class="md-header__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Roberts Lab Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Genomic Resources
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6m0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4M7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Roberts Lab Handbook" class="md-nav__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Roberts Lab Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Lab Culture
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Lab Culture
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Onboarding/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Onboarding
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Code-of-Conduct/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Code of Conduct
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Environment-and-Expectations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Environment and Expectations
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Communication/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Communication
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Notebooks/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Notebooks
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../DEI-Resources/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DEI Resources
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../External-Communication-and-Funding/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    External Communication
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Safety/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Safety
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Undergraduate-contract/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Undergraduate Contract
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../How-to-Grad-School-Slides/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How to Grad School (Outreach Slides)
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Computing and Code
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Computing and Code
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Best-Practices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Best Practices
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Agentic-Coding-Tools/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Agentic Coding Tools
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../code_Snippets/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Code Snippets
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Hardware/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Computing Hardware
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Software
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_6" >
        
          
          <label class="md-nav__link" for="__nav_3_6" id="__nav_3_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Klone Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_6">
            <span class="md-nav__icon md-icon"></span>
            Klone Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_quick-start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quick Start
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_containers/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Containers
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Conda/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Conda
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Adding-a-User/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    New User
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Logging-In/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Logging in
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Node-Types/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Node Types
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Data-Storage-and-System-Organization/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data Storage and System Organization
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_File-Transfers/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    File Transfers
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Installing-Programs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Installing Programs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Running-a-Job/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Running a job
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_RStudio-Server/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RStudio Server
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_VS-Code/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VS Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_7" >
        
          
          <label class="md-nav__link" for="__nav_3_7" id="__nav_3_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Raven Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_7">
            <span class="md-nav__icon md-icon"></span>
            Raven Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../raven_Conda/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Conda
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../raven_RStudio-Server/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RStudio Server
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Databases and Catalogs
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Databases and Catalogs
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Inventory/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Inventory
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Pubathon/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pubathon
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Digital-Media/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Digital Media
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Experiment-Database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Experiment Database
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Genomic Resources
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Genomic Resources
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#acropora-pulchra" class="md-nav__link">
    <span class="md-ellipsis">
      Acropora pulchra
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Acropora pulchra">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bisulfite-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Bisulfite Genome
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#chionoecetes-bairdi" class="md-nav__link">
    <span class="md-ellipsis">
      Chionoecetes bairdi
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cgoreaui" class="md-nav__link">
    <span class="md-ellipsis">
      C.goreaui
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#crassostrea-gigas-cgigas_uk_roslin_v1" class="md-nav__link">
    <span class="md-ellipsis">
      Crassostrea gigas - cgigas_uk_roslin_v1
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#crassostrea-gigas-oyster_v9" class="md-nav__link">
    <span class="md-ellipsis">
      Crassostrea gigas - oyster_v9
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#crassostrea-virginica" class="md-nav__link">
    <span class="md-ellipsis">
      Crassostrea virginica
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gadus-macrocephalus-pacific-cod" class="md-nav__link">
    <span class="md-ellipsis">
      Gadus macrocephalus (Pacific cod)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hematodinium-sp-host-chionoecetes-bairdi" class="md-nav__link">
    <span class="md-ellipsis">
      Hematodinium sp. (Host: Chionoecetes bairdi)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metacarcinus-magister-cancer-magister" class="md-nav__link">
    <span class="md-ellipsis">
      Metacarcinus magister (Cancer magister)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#montipora-capitata" class="md-nav__link">
    <span class="md-ellipsis">
      Montipora capitata
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mytilus-trossulus" class="md-nav__link">
    <span class="md-ellipsis">
      Mytilus trossulus
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ostrea-lurida" class="md-nav__link">
    <span class="md-ellipsis">
      Ostrea lurida
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ostrea-chil" class="md-nav__link">
    <span class="md-ellipsis">
      Ostrea chil
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#panopea-generosa" class="md-nav__link">
    <span class="md-ellipsis">
      Panopea generosa
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pocillipora-acuta" class="md-nav__link">
    <span class="md-ellipsis">
      Pocillipora acuta
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pocillopora-meandrina" class="md-nav__link">
    <span class="md-ellipsis">
      Pocillopora meandrina
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pocillipora-verrucosa" class="md-nav__link">
    <span class="md-ellipsis">
      Pocillipora verrucosa
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycnopodia-helianthodes" class="md-nav__link">
    <span class="md-ellipsis">
      Pycnopodia helianthodes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qpx" class="md-nav__link">
    <span class="md-ellipsis">
      QPX
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#salvelinus-namaycush-lake-trout" class="md-nav__link">
    <span class="md-ellipsis">
      Salvelinus namaycush (lake trout)
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../music/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Spotify Playlists
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Protocols
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Purchasing-and-Reimbursement/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Purchasing and Reimbursement
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Data-Management/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data-Management
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
        
          
          <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Lab Protocols
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5_3">
            <span class="md-nav__icon md-icon"></span>
            Lab Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/protocol-bullet_blender/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bullet Blender
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/protocol-qPCR/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    qPCR
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/resazurin-assay/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Resazurin Assay
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/protocol-reverse_transcription/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reverse Transcription
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Chemical-Standard-Operating-Protocols/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Chemical SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Alarm-Information/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Alarm Information
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bioinformatic Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Bioinformatic Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Basics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Basics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio_DNA-methylation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DNA Methylation Analysis
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Gene-expression/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Gene Expression
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Transcriptome-assembly/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transcriptome Assembly
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#acropora-pulchra" class="md-nav__link">
    <span class="md-ellipsis">
      Acropora pulchra
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Acropora pulchra">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bisulfite-genome" class="md-nav__link">
    <span class="md-ellipsis">
      Bisulfite Genome
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#chionoecetes-bairdi" class="md-nav__link">
    <span class="md-ellipsis">
      Chionoecetes bairdi
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cgoreaui" class="md-nav__link">
    <span class="md-ellipsis">
      C.goreaui
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#crassostrea-gigas-cgigas_uk_roslin_v1" class="md-nav__link">
    <span class="md-ellipsis">
      Crassostrea gigas - cgigas_uk_roslin_v1
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#crassostrea-gigas-oyster_v9" class="md-nav__link">
    <span class="md-ellipsis">
      Crassostrea gigas - oyster_v9
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#crassostrea-virginica" class="md-nav__link">
    <span class="md-ellipsis">
      Crassostrea virginica
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gadus-macrocephalus-pacific-cod" class="md-nav__link">
    <span class="md-ellipsis">
      Gadus macrocephalus (Pacific cod)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hematodinium-sp-host-chionoecetes-bairdi" class="md-nav__link">
    <span class="md-ellipsis">
      Hematodinium sp. (Host: Chionoecetes bairdi)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metacarcinus-magister-cancer-magister" class="md-nav__link">
    <span class="md-ellipsis">
      Metacarcinus magister (Cancer magister)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#montipora-capitata" class="md-nav__link">
    <span class="md-ellipsis">
      Montipora capitata
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mytilus-trossulus" class="md-nav__link">
    <span class="md-ellipsis">
      Mytilus trossulus
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ostrea-lurida" class="md-nav__link">
    <span class="md-ellipsis">
      Ostrea lurida
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ostrea-chil" class="md-nav__link">
    <span class="md-ellipsis">
      Ostrea chil
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#panopea-generosa" class="md-nav__link">
    <span class="md-ellipsis">
      Panopea generosa
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pocillipora-acuta" class="md-nav__link">
    <span class="md-ellipsis">
      Pocillipora acuta
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pocillopora-meandrina" class="md-nav__link">
    <span class="md-ellipsis">
      Pocillopora meandrina
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pocillipora-verrucosa" class="md-nav__link">
    <span class="md-ellipsis">
      Pocillipora verrucosa
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pycnopodia-helianthodes" class="md-nav__link">
    <span class="md-ellipsis">
      Pycnopodia helianthodes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qpx" class="md-nav__link">
    <span class="md-ellipsis">
      QPX
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#salvelinus-namaycush-lake-trout" class="md-nav__link">
    <span class="md-ellipsis">
      Salvelinus namaycush (lake trout)
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/RobertsLab/resources/blob/master/docs/Genomic-Resources.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75z"/></svg>
    </a>
  
  


  <h1>Genomic Resources</h1>

<p>Here we try to compile genomic resources such that they are readily available and somewhat described. An effort will be made to keep respective index files alongside so these files can be directly used in IGV <em>etc</em>.</p>
<p><strong>Related Resources</strong> - Archived Versions of this page - <a href="http://htmlpreview.github.io/?https://github.com/RobertsLab/resources/blob/master/archive/Genomic-Resources.091419.html">091319</a>;</p>
<ul>
<li><a href="https://docs.google.com/spreadsheets/d/1_XqIOPVHSBVGscnjzDSWUeRL7HUHXfaHxVzec-I-8Xk/edit?usp=sharing">Nightingales (Google Sheet)</a> : Database of all raw high-throughput sequencing data</li>
</ul>
<hr />
<h2 id="acropora-pulchra">Acropora pulchra</h2>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/gitrepos/urol-e5/timeseries_molecular/D-Apul/data/Apulchra-genome.fa">Apulchra-genome.fa</a> (504MB): <code>https://gannet.fish.washington.edu/gitrepos/urol-e5/timeseries_molecular/D-Apul/data/Apulchra-genome.fa</code></p>
</li>
<li>
<p>MD5 = <code>1c500bff4aa07f8a097adf79260bf141</code></p>
</li>
<li>
<p>FastA index (<code>samtoold faidx</code>)</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/gitrepos/urol-e5/timeseries_molecular/D-Apul/data/Apulchra-genome.fa.fai">Apulchra-genome.fa.fai</a></li>
</ul>
</li>
<li>
<p>Origin of this genome comes from this <a href="https://github.com/urol-e5/timeseries_molecular/issues/1#issue-2564344644">GitHub Issue</a>.</p>
</li>
</ul>
<h3 id="bisulfite-genome">Bisulfite Genome</h3>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/gitrepos/urol-e5/timeseries_molecular/D-Apul/data/Bisulfite_Genome.tar.gz">Bisulfite_Genome.tar.gz</a> (1.3GB): <code>https://gannet.fish.washington.edu/gitrepos/urol-e5/timeseries_molecular/D-Apul/data/Bisulfite_Genome.tar.gz</code></p>
</li>
<li>
<p>MD5 = <code>399a196fc1dfc202a9f084a8fb342dd9</code></p>
</li>
<li>Gzipped tarball of bisulfite genome for use with Bismark.</li>
<li>Creation details <a href="https://github.com/urol-e5/timeseries_molecular/blob/f8b2e0100ecaa9c0d008f7466cabdf54ab261de8/D-Apul/code/00.21-D-Apul-BS-genome.md">here</a>.</li>
</ul>
<h2 id="chionoecetes-bairdi">Chionoecetes bairdi</h2>
<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_genome_v1.01.fasta">cbai_genome_v1.01.fasta</a> (18MB)<ul>
<li>MD5 = <code>5a08d8b0651484e3ff75fcf032804596</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-09-24-Assembly-Assessment---BUSCO-C.bairdi-Genome-v1.01-on-Mox/index.html">BUSCOs</a>:
<code>C:0.4%[S:0.3%,D:0.1%],F:0.2%,M:99.4%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_genome_v1.01.fasta.fai">cbai_genome_v1.01.fasta.fai</a></li>
</ul>
</li>
<li>Assembly from <a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-09-23-Data-Wrangling---Subsetting-cbai_genome_v1.0-Assembly-with-faidx/index.html">20200923</a><ul>
<li>Q7-filtered NanoPore data. Includes <em>Hematodinium</em>-infected sample.</li>
<li>Subset of <code>cbai_genome_v1.0.fasta</code> with contigs &gt;1000bp</li>
</ul>
</li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_genome_v1.0.fasta">cbai_genome_v1.0.fasta</a> (19MB)<ul>
<li>MD5 = <code>2f3b651bb0b875b0287e71e315cad59a</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/2020/09/18/Assembly-Assessment-BUSCO-C.bairdi-Genome-v1.0-on-Mox.html">BUSCOs</a>:
<code>C:0.4%[S:0.3%,D:0.1%],F:0.3%,M:99.3%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_genome_v1.0.fasta.fai">cbai_genome_v1.0.fasta.fai</a></li>
</ul>
</li>
<li>Assembly from <a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-09-17-Genome-Assembly---C.bairdi---cbai_v1.0---Using-All-NanoPore-Data-With-Flye-on-Mox/index.html">20200917</a><ul>
<li>Q7-filtered NanoPore data. Includes <em>Hematodinium</em>-infected sample.
<a href="https://docs.google.com/spreadsheets/d/1XAgU_xQKJjWk4ThJHn1wLDtPuW6X7s6Jjh_373bMc0U/edit?usp=sharing">Assembly Stats Table (Google Sheet)</a>
<a href="https://docs.google.com/spreadsheets/d/1d17yg5F5gKKC66O8QkTIlPxljJeuX7ZsG46pkBr1lNQ/edit#gid=0">RNA-seq sample list</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210317_cbai_trinity_RNAseq_transcriptome-v4.0/cbai_transcriptome_v4.0.fasta_trinity_out_dir/cbai_transcriptome_v4.0.fasta">cbai_transcriptome_v4.0.fasta</a><ul>
<li>MD5 = <code>6450d6f5650bfb5f910a5f42eef94913</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/2021/03/17/Transcriptome-Assessment-BUSCO-Metazoa-on-C.bairdi-Transcriptome-v4.0-on-Mox.html">BUSCOs</a>: <code>C:73.8%[S:45.8%,D:28.0%],F:7.9%,M:18.3%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210317_cbai_trinity_RNAseq_transcriptome-v4.0/cbai_transcriptome_v4.0.fasta_trinity_out_dir/cbai_transcriptome_v4.0.fasta.fai">cbai_transcriptome_v4.0.fasta.fai</a></li>
</ul>
</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210318_cbai_diamond_blastx_transcriptome-v4.0/cbai_transcriptome_v4.0.blastx.outfmt6">BLASTx annotation</a> (outfmt6)</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20210318_cbai_trinotate_transcriptome-v4.0/20210318.cbai_transcriptome_v4.0.fasta.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-GW, 2020-UW BLASTx against NCBI <em>C.opilio</em> genome.</li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v3.1.fasta">cbai_transcriptome_v3.1.fasta</a><ul>
<li>MD5 = <code>aeec8ffbf8fa44fb1750caee6abaf68a</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-06-05-Transcriptome-Assessment---BUSCO-Metazoa-on-C.bairdi-Transcriptome-v3.1/">BUSCOs</a>: <code>C:96.5%[S:40.3%,D:56.2%],F:2.2%,M:1.3%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v3.1.fasta.fai">cbai_transcriptome_v3.1.fasta.fai</a></li>
</ul>
</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200608_cbai_diamond_blastx_v2.1_v3.1/cbai_transcriptome_v3.1.blastx.outfmt6">BLASTx annotation</a> (outfmt6)</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200828_cbai_trinotate_transcriptome-v3.1/20200828.cbai_transcriptome_v3.1.fasta.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-UW with <em>non_Alveolata</em>. <a href="https://robertslab.github.io/sams-notebook/2020/06/05/Sequence-Extractions-C.bairdi-Transcriptomes-v2.0-and-v3.0-Excluding-Alveolata-with-MEGAN6-on-Swoose.html">Derived from <code>cbai_transcriptome_v3.0.fasta</code></a></li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v3.0.fasta">cbai_transcriptome_v3.0.fasta</a><ul>
<li>Assembly from <a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-05-18-Transcriptome-Assembly---C.bairdi-All-Pooled-RNAseq-Data-Without-Taxonomic-Filters-with-Trinity-on-Mox/index.html">20200518</a></li>
<li>MD5 = <code>5516789cbad5fa9009c3566003557875</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-05-19-Transcriptome-Assessment---BUSCO-Metazoa-on-C.bairdi-Transcriptome-v3.0/index.html">BUSCOs</a>: <code>C:97.6%[S:39.1%,D:58.5%],F:1.6%,M:0.8%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v3.0.fasta.fai">cbai_transcriptome_v3.0.fasta.fai</a></li>
</ul>
</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200519_cbai_diamond_blastx_transcriptome_v3.0/20200518.C_bairdi.Trinity.blastx.outfmt6">BLASTx annotation</a> (outfmt6)</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200526_cbai_trinotate_transcriptome-v3.0/20200526.cbai_transcriptome_v3.0.fasta.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-UW with <em>no taxonomic filter</em>.</li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v2.1.fasta">cbai_transcriptome_v2.1.fasta</a><ul>
<li>MD5 = <code>1fb788175f9bb7cd5145370a399ae857</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/2020/06/05/Transcriptome-Assessment-BUSCO-Metazoa-on-C.bairdi-Transcriptome-v2.1.html">BUSCOs</a>: <code>C:98.3%[S:25.2%,D:73.1%],F:1.4%,M:0.3%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v2.1.fasta.fai">cbai_transcriptome_v2.1.fasta.fai</a></li>
</ul>
</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200608_cbai_diamond_blastx_v2.1_v3.1/cbai_transcriptome_v2.1.blastx.outfmt6">BLASTx annotation</a></li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200831_cbai_trinotate_transcriptome-v2.1/20200831.cbai_transcriptome_v2.1.fasta.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-GW, 2020-UW with <em>non_Alveolata</em>. <a href="https://robertslab.github.io/sams-notebook/2020/06/05/Sequence-Extractions-C.bairdi-Transcriptomes-v2.0-and-v3.0-Excluding-Alveolata-with-MEGAN6-on-Swoose.html">Derived from <code>cbai_transcriptome_v2.0.fasta</code></a></li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v2.0.fasta">cbai_transcriptome_v2.0.fasta</a><ul>
<li>Also referred to as <a href="https://robertslab.github.io/sams-notebook/2020/05/02/Transcriptome-Assembly-C.bairdi-All-RNAseq-Data-Without-Taxonomic-Filters-with-Trinity-on-Mox.html"><code>20200507.C_bairdi.Trinity.fasta</code></a>.</li>
<li>MD5 = <code>01adbd54298495c147767b19ee5c0de9</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/2020/05/08/Transcriptome-Assessment-BUSCO-Metazoa-on-C.bairdi-v2.0-Transcriptome.html">BUSCOs</a>: <code>C:98.8%[S:24.9%,D:73.9%],F:0.9%,M:0.3%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v2.0.fasta.fai">cbai_transcriptome_v2.0.fasta.fai</a></li>
</ul>
</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200508_cbai_diamond_blastx_transcriptome-v2.0/20200507.C_bairdi.Trinity.blastx.outfmt6">BLASTx annotation</a></li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200513_cbai_trinotate_transcriptome-v2.0/20200513.cbai.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-GW, 2020-UW with <em>no taxonomic filter</em>.</li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.7.fasta">cbai_transcriptome_v1.7.fasta</a><ul>
<li>MD5 = <code>032d1f81c7744736ebeefe7f63ed6d95</code></li>
<li>Assembly from <a href="https://robertslab.github.io/sams-notebook/2020/05/27/Transcriptome-Assembly-C.bairdi-All-Pooled-Arthropoda-only-RNAseq-Data-with-Trinity-on-Mox.html">20200527</a></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.7.fasta.fai">cbai_transcriptome_v1.7.fasta.fai</a> :
  <code>https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.7.fasta.fai</code></li>
</ul>
</li>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-05-27-Transcriptome-Assessment---BUSCO-Metazoa-on-C.bairdi-Transcriptome-v1.7/">BUSCOs</a>: <code>C:86.7%[S:66.5%,D:20.2%],F:8.2%,M:5.1%,n:978</code></li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200527_cbai_diamond_blastx_transcriptome_v1.7/cbai_transcriptome_v1.7.blastx.outfmt6">BLASTx Annotation</a> (outfmt6)</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200529_cbai_trinotate_transcriptome-v1.7/20200616.cbai_transcriptome_v1.7.fasta.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-UW with <em>Arthropoda</em> only reads.</li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.6.fasta">cbai_transcriptome_v1.6.fasta</a><ul>
<li>MD5 = <code>46d77ce86cdbbcac26bf1a6cb820651e</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.6.fasta.fai">cbai_transcriptome_v1.6.fasta.fai</a> :
<code>https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.6.fasta.fai</code></li>
</ul>
</li>
<li><a href="https://robertslab.github.io/sams-notebook/2020/05/19/Transcriptome-Assessment-BUSCO-Metazoa-on-C.bairdi-Transcriptome-v1.6.html">BUSCOs</a>: <code>C:91.7%[S:62.6%,D:29.1%],F:6.2%,M:2.1%,n:978</code></li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200519_cbai_diamond_blastx_transcriptome_v1.6/cbai_transcriptome_v1.6.blastx.outfmt6">BLASTx Annotation</a> (outfmt6)</li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200520_cbai_trinotate_transcriptome-v1.6/20200616.cbai_transcriptome_v1.6.fasta.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-GW, 2020-UW with <em>Arthropoda</em> only reads.</li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.5.fasta">cbai_transcriptome_v1.5.fasta</a><ul>
<li>MD5 = <code>e61d68c45728ffbb91e3d34c087d9aa9</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-04-07-Transcriptome-Assessment---BUSCO-Metazoa-on-C.bairdi-MEGAN-Transcriptome/index.html">BUSCOs</a>: C:91.8%[S:64.0%,D:27.8%],F:5.9%,M:2.3%,n:978</li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.5.fasta.fai">cbai_transcriptome_v1.5.fasta.fai</a> :
  <code>https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.5.fasta.fai</code></li>
</ul>
</li>
<li>Updated assembly from <a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-03-30-Transcriptome-Assembly---C.bairdi-with-MEGAN6-Taxonomy-specific-Reads-with-Trinity-on-Mox/">20200330</a>. Also referred to as <code>20200408.C_bairdi.megan.Trinity.fasta</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-04-08-Transcriptome-Annotation---C.bairdi-MEGAN-Trinity-Assembly-Using-DIAMOND-BLASTx-on-Mox/index.html">BLASTx Annotation</a></li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200409_cbai_trinotate_megan/20200409.cbai.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019, 2020-GW with Arthropoda only reads.</li>
</ul>
</li>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.0.fasta">cbai_transcriptome_v1.0.fasta</a><ul>
<li>MD5 = <code>fb28a203154b44b67ec2e2476d96d326</code></li>
<li><a href="https://robertslab.github.io/sams-notebook/2020/02/07/Transcriptome-Assessment-BUSCO-Metazoa-on-C.bairdi-MEGAN-Transcriptome.html">BUSCOs</a>: <code>C:85.5%[S:64.7%,D:20.8%],F:9.3%,M:5.2%,n:978</code></li>
<li>FastA index (<code>samtools faidx</code>)<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.0.fasta.fai">cbai_transcriptome_v1.0.fasta.fai</a> :
  <code>https://owl.fish.washington.edu/halfshell/genomic-databank/cbai_transcriptome_v1.0.fasta.fai</code></li>
</ul>
</li>
<li>Initial Trinity assembly from <a href="https://robertslab.github.io/sams-notebook/2020/01/22/Transcriptome-Assembly-C.bairdi-with-MEGAN6-Taxonomy-specific-Reads-with-Trinity-on-Mox.html">20200122</a></li>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-01-23-Transcriptome-Annotation---C.bairdi-MEGAN-Trinity-Assembly-Using-DIAMOND-BLASTx-on-Mox/index.html">BLASTx Annotation</a></li>
<li><a href="https://gannet.fish.washington.edu/Atumefaciens/20200126_cbai_trinotate_megan/20200126.cbai.trinotate.go_annotations.txt">GO Terms Annotation</a> (Trinotate)</li>
<li>internal short-hand: includes 2018, 2019 with <em>Arthropoda</em> only reads.
editor_options: 
  markdown: 
wrap: 72</li>
</ul>
</li>
</ul>
<hr />
<h2 id="cgoreaui"><em>C.goreaui</em></h2>
<p>Genomes:</p>
<ul>
<li>
<p><code>/volume1/web/halfshell/genomic-databank/Cladocopium_goreaui_genome_fa</code> (1.1GB)</p>
<ul>
<li>
<p>MD5 checksum: <code>eb4a1a7ac2fc0cbc6f5c178240beb932</code></p>
</li>
<li>
<p>Downloaded 20230216: <a href="https://espace.library.uq.edu.au/view/UQ:fba3259">https://espace.library.uq.edu.au/view/UQ:fba3259</a></p>
</li>
<li>
<p>Access to the genome requires agreeing to some licensing provisions (primarily the requirement to cite the publication whenever the genome is used), so we will not be providing any public links to the file.</p>
</li>
<li>
<p><a href="https://doi.org/10.48610/fba3259">Chen et. al, 2022</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Indexes (<a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>):</p>
<ul>
<li>
<p><a href="">``</a> (tarball gzip; 563MB)</p>
<ul>
<li>
<p>MD5 checksum: ``</p>
</li>
<li>
<p>Needs to be unpacked before use!</p>
</li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks</p>
<ul>
<li>
<p><code>/volume1/web/halfshell/genomic-databank/Cladocopium_goreaui_genes_gff3</code> (225MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>ab47babf331507b9284e9d35406aefac</code></p>
</li>
<li>
<p>Downloaded 20230216: <a href="https://espace.library.uq.edu.au/view/UQ:fba3259">https://espace.library.uq.edu.au/view/UQ:fba3259</a></p>
</li>
<li>
<p>Access to the GFF requires agreeing to some licensing provisions (primarily the requirement to cite the publication whenever the genome is used), so we will not be providing any public links to the file.</p>
</li>
<li>
<p><a href="https://doi.org/10.48610/fba3259">Chen et. al, 2022</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230217-cgor-gff_to_gtf/Cladocopium_goreaui_genes_gff3.gtf"><code>Cladocopium_goreaui_genes_gff3.gtf</code></a> (197MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>97e69a850faf2e6d9b60df828ad02671</code></p>
</li>
<li>
<p>Created 20230217: <a href="https://robertslab.github.io/sams-notebook/posts/2023/2023-02-17-Data-Wrangling---C.goreaui-Genome-GFF-to-GTF-Using-gffread/index.html">Data-Wrangling-C.goreaui-Genome-GFF-to-GTF-Using-gffread</a></p>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="crassostrea-gigas-cgigas_uk_roslin_v1"><em>Crassostrea gigas</em> - cgigas_uk_roslin_v1</h2>
<ul>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_902806645.1/">NCBI Assembly GCF_902806645.1</a></p>
</li>
<li>
<p><a href="https://doi.org/10.1093/gigascience/giab020">A chromosome-level genome assembly for the Pacific oyster Crassostrea gigas</a></p>
</li>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Crassostrea_gigas/102/">NCBI Crassostrea gigas Annotation Release 102</a></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank//cgigas_uk_roslin_v1_fuzznuc_CGmotif.gff">cgigas_uk_roslin_v1_fuzznuc_CGmotif.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_fuzznuc_CGmotif.gff</code> (CG motif track)</p>
</li>
</ul>
<p>Genome assembly with mitochondrial DNA included: - <a href="https://gannet.fish.washington.edu/panopea/Cg-roslin/cgigas_uk_roslin_v1_genomic-mito.fa">cgigas_uk_roslin_v1_genomic-mito.fa</a>: <code>https://gannet.fish.washington.edu/panopea/Cg-roslin/cgigas_uk_roslin_v1_genomic-mito.fa</code></p>
<ul>
<li><a href="https://gannet.fish.washington.edu/panopea/Cg-roslin/cgigas_uk_roslin_v1_genomic-mito.fa.fai">cgigas_uk_roslin_v1_genomic-mito.fa.fai</a>: <code>https://gannet.fish.washington.edu/panopea/Cg-roslin/cgigas_uk_roslin_v1_genomic-mito.fa.fai</code></li>
</ul>
<p>Genome feature tracks generated from the <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/902/806/645/GCF_902806645.1_cgigas_uk_roslin_v1/">NCBI RefSeq link</a> in <a href="https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/08-Generating-Genome-Feature-Tracks.ipynb">this Jupyter notebook</a></p>
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_gene.gff">cgigas_uk_roslin_v1_gene.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_gene.gff</code></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/panopea/Cg-roslin/GCF_902806645.1_cgigas_uk_roslin_v1_genomic-mito.gtf">GCF_902806645.1_cgigas_uk_roslin_v1_genomic-mito.gtf</a>: <code>https://gannet.fish.washington.edu/panopea/Cg-roslin/GCF_902806645.1_cgigas_uk_roslin_v1_genomic-mito.gtf</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_mRNA.gff">cgigas_uk_roslin_v1_mRNA.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_mRNA.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_CDS.gff">cgigas_uk_roslin_v1_CDS.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_CDS.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_nonCDS.bed">cgigas_uk_roslin_v1_nonCDS.bed</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_nonCDS.bed</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_exon.gff">cgigas_uk_roslin_v1_exon.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_exon.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_exonUTR.gff">cgigas_uk_roslin_v1_exonUTR.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_exonUTR.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_intron.bed">cgigas_uk_roslin_v1_intron.bed</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_intron.bed</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_intergenic.bed">cgigas_uk_roslin_v1_intergenic.bed</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_intergenic.bed</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_flanks.gff">cgigas_uk_roslin_v1_flanks.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_flanks.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_upstream.gff">cgigas_uk_roslin_v1_upstream.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_upstream.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_downstream.gff">cgigas_uk_roslin_v1_downstream.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_downstream.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_lncRNA.gff">cgigas_uk_roslin_v1_lncRNA.gff</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_lncRNA.gff</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_rm.te.bed">cgigas_uk_roslin_v1_rm.te.bed</a>: <code>http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_rm.te.bed</code></p>
</li>
</ul>
<h2 id="crassostrea-gigas-oyster_v9"><em>Crassostrea gigas</em> - oyster_v9</h2>
<p><strong>Related Resources</strong></p>
<ul>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/datasets/genomes/?txid=29159">NCBI Datasets</a></p>
</li>
<li>
<p><a href="https://figshare.com/articles/Compilation_of_DNA_Methylation_Genome_Feature_Tracks_Crassostrea_gigas_/1456226">Compilation of DNA Methylation Genome Feature Tracks (<em>Crassostrea gigas</em>)</a> <em>circa 2015</em></p>
</li>
<li>
<p><a href="http://onsnetwork.org/halfshell/2015/02/27/re-defining-cgigas-canonical-features/">Re-defining Cgigas Canonical features</a> <em>circa 2015</em></p>
</li>
<li>
<p><a href="http://gigaton.sigenae.org/ngspipelines/#!/NGSpipelines/Crassostrea%20gigas%20-%20GIGATON">Gigaton</a></p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/resources/blob/master/html/tjgr.markdown">TJGR</a></p>
</li>
</ul>
<p>Genome:</p>
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.dna_sm.toplevel.fa">Crassostrea_gigas.oyster_v9.dna_sm.toplevel.fa</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.dna_sm.toplevel.fa</code></p>
<ul>
<li>
<p>MD5 = 6de9d1239eb10ea0545bed6c4e746d6c</p>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>) : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.dna_sm.toplevel.fa.fai</code></p>
</li>
</ul>
</li>
</ul>
<p>Bisulfite Genome:</p>
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.dna_sm.toplevel_bisulfite.tar.gz">Crassostrea_gigas.oyster_v9.dna_sm.toplevel_bisulfite.tar.gz</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.dna_sm.toplevel_bisulfite.tar.gz</code></p>
<ul>
<li>
<p>Gzipped tarball of bisulfite genome for use with Bismark</p>
</li>
<li>
<p>Creation details <a href="https://robertslab.github.io/sams-notebook/2019/02/21/Data-Management-Create-C.gigas-Bisulfite-Genome-with-Bismark-on-Mox.html">here</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks</p>
<ul>
<li>
<p><a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_gene.gff">Cgigas_v9_gene.gff</a> : <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_gene.gff</code></p>
</li>
<li>
<p><a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_exon.gff">Cgigas_v9_exon.gff</a> : <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_exon.gff</code></p>
</li>
<li>
<p><a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_intron.gff">Cgigas_v9_intron.gff</a> : <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_intron.gff</code></p>
</li>
<li>
<p><a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_TE.gff">Cgigas_v9_TE.gff</a> : <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_TE.gff</code></p>
<ul>
<li>Contains <a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_TE-TANDEMREPEAT.gff">Tandem Repeats</a> and <a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_TE-WUBLASTX.gff">wublastx</a> features.</li>
</ul>
</li>
<li>
<p><a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_CG.gff">Cgigas_v9_CG.gff</a> : <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_CG.gff</code></p>
<ul>
<li>index: <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_CG.gff.idx</code></li>
</ul>
</li>
<li>
<p><a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_1k5p_gene_promoter.gff">Cgigas_v9_1k5p_gene_promoter.gff</a> : <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_1k5p_gene_promoter.gff</code></p>
</li>
<li>
<p><a href="https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_COMP_gene_prom_TE.bed">Cgigas_v9_COMP_gene_prom_TE.bed</a> : <code>https://eagle.fish.washington.edu/trilobite/Crassostrea_gigas_v9_tracks/Cgigas_v9_COMP_gene_prom_TE.bed</code></p>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.40.gff3">Crassostrea_gigas.oyster_v9.40.gff3</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.40.gff3</code></p>
<ul>
<li>MD5 = 90a747fbc94a0a9225c43f75cc40b9db</li>
</ul>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.40.abinitio.gff3">Crassostrea_gigas.oyster_v9.40.abinitio.gff3</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.40.abinitio.gff3</code></p>
<ul>
<li>MD5 = c2a8c388f5a8afb22a115d61dee3dda0</li>
</ul>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.40_mRNA.gff3">Crassostrea_gigas.oyster_v9.40_mRNA.gff3</a></p>
<ul>
<li><code>grep "mRNA" Crassostrea_gigas.oyster_v9.40.gff3 &gt; Crassostrea_gigas.oyster_v9.40_mRNA.gff3</code></li>
</ul>
</li>
</ul>
<hr />
<h2 id="crassostrea-virginica"><em>Crassostrea virginica</em></h2>
<p><a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/022/765/GCF_002022765.2_C_virginica-3.0/">NCBI FTP</a></p>
<p>Genomes:</p>
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Cvirginica_v300.fa">Cvirginica_v300.fa</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Cvirginica_v300.fa</code></p>
<ul>
<li>
<p>MD5 = f9135e323583dc77fc726e9df2677a32</p>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>)</p>
<ul>
<li><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Cvirginica_v300.fa.fai">Cvirginica_v300.fa.fai</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Cvirginica_v300.fa.fai</code></li>
</ul>
</li>
</ul>
</li>
<li>
<p><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/022/765/GCF_002022765.2_C_virginica-3.0/GCF_002022765.2_C_virginica-3.0_genomic.fna.gz">GCF_002022765.2_C_virginica-3.0_genomic.fna.gz</a> : <code>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/022/765/GCF_002022765.2_C_virginica-3.0/GCF_002022765.2_C_virginica-3.0_genomic.fna.gz</code></p>
<ul>
<li>compressed version of <code>Cvirginica_v300.fa</code> (same files)</li>
</ul>
</li>
</ul>
<p>Annotations:</p>
<ul>
<li>
<p>Coding sequences (CDS):</p>
<ul>
<li>
<p><a href="https://github.com/sr320/ceabigr/blob/main/output/17-Swiss-Prot-Annotation/Cvir_cds-geneID-SPID.tab"><code>Cvir_cds-geneID-SPID.tab</code></a></p>
<ul>
<li>
<p>MD5 = <code>0ee045d4e4702094798da26c2dd6fca3</code></p>
</li>
<li>
<p>Gene with corresponsing SPID.</p>
</li>
<li>
<p><a href="https://github.com/sr320/ceabigr/blob/main/code/17-Swiss-Prot-Annotation.Rmd">17-Swiss-Prot-Annotation.Rmd</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://github.com/sr320/ceabigr/tree/main/output/17.1-GO-and-GOslim-CDS-Annotation/Cvir-CDS-GOslim.BP_per_gene.tab"><code>Cvir-CDS-GOslim.BP_per_gene.tab</code></a></p>
<ul>
<li>
<p>MD5 = <code>73eac024baeedf06e28121f201525945</code></p>
</li>
<li>
<p>Biological process GOslims/terms per gene</p>
</li>
<li>
<p><a href="https://github.com/sr320/ceabigr/blob/main/code/17.1-GO-and-GOslim-CDS-Annotation.Rmd">17.1-GO-and-GOslim-CDS-Annotation.Rmd</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://github.com/sr320/ceabigr/tree/main/output/17.1-GO-and-GOslim-CDS-Annotation/Cvir-CDS-uniprot-full.tsv"><code>Cvir-CDS-uniprot-full.tsv</code></a></p>
<ul>
<li>
<p>MD5 = <code>80dd6338e42f21df120bf0cc22f95eab</code></p>
</li>
<li>
<p>Tab-delimited output file from UniProt API retrieval. Columns:</p>
<ul>
<li><code>Entry</code></li>
<li><code>Reviewed</code></li>
<li><code>Entry Name</code></li>
<li><code>Protein names</code></li>
<li><code>Gene Names</code></li>
<li><code>Organism</code></li>
<li><code>Length</code></li>
<li><code>Gene Ontology (biological process)</code></li>
<li><code>Gene Ontology (cellular component)</code></li>
<li><code>Gene Ontology (GO)</code></li>
<li><code>Gene Ontology (molecular function)</code></li>
<li><code>Gene Ontology IDs</code></li>
</ul>
</li>
<li>
<p><a href="https://github.com/sr320/ceabigr/blob/main/code/17.1-GO-and-GOslim-CDS-Annotation.Rmd">17.1-GO-and-GOslim-CDS-Annotation.Rmd</a></p>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>Bisulfite Genomes:</p>
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Cvirginica_v300_bisulfite.tar.gz">Cvirginica_v300_bisulfite.tar.gz</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Cvirginica_v300_bisulfite.tar.gz</code></p>
<ul>
<li>
<p>Gzipped tarball of bisulfite genome for use with Bismark</p>
</li>
<li>
<p>Creation details <a href="https://robertslab.github.io/sams-notebook/posts/2019/2019-02-21-Data-Management---Create-C.virginica-Bisulfite-Genome-with-Bismark-on-Mox/index.html">here</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks</p>
<ul>
<li>
<p><a href="http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_Gnomon_mRNA.gff3">C_virginica-3.0_Gnomon_mRNA.gff3</a> : <code>http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_Gnomon_mRNA.gff3</code></p>
</li>
<li>
<p><a href="https://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_Gnomon_genes.bed">C_virginica-3.0_Gnomon_genes.bed</a> : <code>https://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_Gnomon_genes.bed</code></p>
<ul>
<li>
<p>MD5 = <code>c8f203de591c0608b96f4299c0f847dc</code></p>
</li>
<li>
<p><a href="https://robertslab.github.io/sams-notebook/posts/2021/2021-12-09-Data-Wrangling---C.virginica-NCBI-GCF_002022765.2-GFF-to-Gene-BED-File/">Notebook entry</a></p>
</li>
</ul>
</li>
<li>
<p><a href="http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_Gnomon_exon.bed">C_virginica-3.0_Gnomon_exon.bed</a> : <code>http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_Gnomon_exon.bed</code></p>
</li>
<li>
<p><a href="http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_intron.bed">C_virginica-3.0_intron.bed</a> : <code>http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_intron.bed</code></p>
</li>
<li>
<p><a href="http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_CG-motif.bed">C_virginica-3.0_CG-motif.bed</a> : <code>http://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_CG-motif.bed</code></p>
<ul>
<li>
<p>MD5 = <code>f88c171bccf45a6f3afcf455b6be810f</code></p>
</li>
<li>
<p>Dead link in this Jupyter Notebook obscures details on how this was generated (via Galaxy):</p>
<ul>
<li><a href="https://github.com/sr320/nb-2018/blob/master/C_virginica/22-CG-track.ipynb">https://github.com/sr320/nb-2018/blob/master/C_virginica/22-CG-track.ipynb</a></li>
</ul>
</li>
</ul>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/C_virginica-3.0_TE-all.gff">C_virginica-3.0_TE-all.gff</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/C_virginica-3.0_TE-all.gff</code></p>
<ul>
<li>
<p>MD5 = d0d81fc6cf7525bc2c61984bee23521b</p>
</li>
<li>
<p><a href="https://robertslab.github.io/sams-notebook/posts/2018/2018-08-28-transposable-element-mapping-crassostrea-virginica-genome-cvirginica_v300-using-repeatmasker-4-07/">Details</a></p>
</li>
</ul>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/C_virginica-3.0_TE-Cg.gff">C_virginica-3.0_TE-Cg.gff</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/C_virginica-3.0_TE-Cg.gff</code></p>
<ul>
<li>
<p>MD5 = 83cd753c171076464fee1165b7e1c6ba</p>
</li>
<li>
<p><a href="https://robertslab.github.io/sams-notebook/posts/2018/2018-08-28-transposable-element-mapping-crassostrea-virginica-genome-cvirginica_v300-using-repeatmasker-4-07/">Details</a></p>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="gadus-macrocephalus-pacific-cod"><em>Gadus macrocephalus</em> (Pacific cod)</h2>
<p><a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_031168955.1/">NCBI</a></p>
<p>Genomes</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCF_031168955.1_ASM3116895v1_genomic.fna"><code>GCF_031168955.1_ASM3116895v1_genomic.fna</code></a> (FastA; 537MB)</p>
<ul>
<li>MD5 = <code>5144890d4eceb0b258d92db3f35c681e</code></li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCF_031168955.1_ASM3116895v1_cds_from_genomic.fna"><code>GCF_031168955.1_ASM3116895v1_cds_from_genomic.fna</code></a> (FastA; 112MB)</p>
<ul>
<li>MD5 = <code>95ef994762e8e4d58c2ce15d312f0aaf</code></li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCF_031168955.1_ASM3116895v1.faa"><code>GCF_031168955.1_ASM3116895v1.faa</code></a> (FastA; 35MB)</p>
<ul>
<li>MD5 = <code>cf58a1ec6408eb8c55aa23a6c06430d9</code></li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCF_031168955.1_ASM3116895v1_rna.fna"><code>GCF_031168955.1_ASM3116895v1_rna.fna</code></a> (FastA; 169MB)</p>
<ul>
<li>MD5 = <code>2a6c7c98982727e688f033a9b236725b</code></li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCF_031168955.1_ASM3116895v1.gff"><code>GCF_031168955.1_ASM3116895v1.gff</code></a> (GFF; 351MB)</p>
</li>
<li>
<p>MD5 = <code>173fb3c159e474391c5c4aa1f7230024</code></p>
</li>
</ul>
<h2 id="hematodinium-sp-host-chionoecetes-bairdi"><em>Hematodinium sp.</em> (Host: <em>Chionoecetes bairdi</em>)</h2>
<p>Transcriptomes</p>
<p><a href="https://docs.google.com/spreadsheets/d/1A81cFdFw5Mlks5DWMmq0-8eVqyTXqmoCsHNWs95N_p4/edit?usp=sharing">Assembly Stats Table (Google Sheet)</a></p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210308_hemat_trinity_v1.6_v1.7/hemat_transcriptome_v1.7.fasta_trinity_out_dir/hemat_transcriptome_v1.7.fasta">hemat_transcriptome_v1.7.fasta</a></p>
<div class="highlight"><pre><span></span><code>- internal short-hand: includes 2018, 2019, 2020-UW with _Alveolata_ only reads.

- MD5 = `f9c8f96a49506e1810ff4004426160d8`

- FastA index (```samtools faidx```)

    - [hemat_transcriptome_v1.7.fasta.fai](https://gannet.fish.washington.edu/Atumefaciens/20210308_hemat_trinity_v1.6_v1.7/hemat_transcriptome_v1.7.fasta_trinity_out_dir/hemat_transcriptome_v1.7.fasta.fai)

- [Notebook entry](https://robertslab.github.io/sams-notebook/2021/03/08/Transcriptome-Assembly-Hematodinium-Transcriptomes-v1.6-and-v1.7-with-Trinity-on-Mox.html)

- BUSCOs: `C:15.0%[S:12.2%,D:2.8%],F:12.3%,M:72.7%,n:978`

    - [Notebook entry](https://robertslab.github.io/sams-notebook/2020/08/14/Transcriptome-Assessment-BUSCO-Metazoa-on-Hematodinium-v1.6-v1.7-v2.1-and-v3.1-on-Mox.html)

- BLASTx Annotation

  - [hemat_transcriptome_v1.7.fasta.blastx.outfmt6](https://gannet.fish.washington.edu/Atumefaciens/20200814_hemat_diamond_blastx_v1.6_v1.7_v2.1_v3.1/hemat_transcriptome_v1.7.fasta.blastx.outfmt6)

  - [Notebook entry](https://robertslab.github.io/sams-notebook/2020/08/14/Transcriptome-Annotation-Hematodinium-Transcriptomes-v1.6-v1.7-v2.1-v3.1-with-DIAMOND-BLASTx-on-Mox.html)

- GO Terms Annotation

  - [20210310.hemat_transcriptome_v1.7.fasta.trinotate.go_annotations.txt](https://gannet.fish.washington.edu/Atumefaciens/20210309_hemat_trinotate_transcriptome-v1.7/20210310.hemat_transcriptome_v1.7.fasta.trinotate.go_annotations.txt) (Trinotate)

  - [Notebook entry](https://robertslab.github.io/sams-notebook/posts/2021/2021-03-09-Transcriptome-Annotation---Trinotate-Hematodinium-v1.7-on-Mox/index.html)
</code></pre></div>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20210308_hemat_trinity_v1.6_v1.7/hemat_transcriptome_v1.6.fasta_trinity_out_dir/hemat_transcriptome_v1.6.fasta">hemat_transcriptome_v1.6.fasta</a></p>
<div class="highlight"><pre><span></span><code>- internal short-hand: includes 2018, 2019, 2020-GW, 2020-UW with _Alveolata_ only reads.

- MD5 = `f9c8f96a49506e1810ff4004426160d8`

- FastA index (```samtools faidx```)

    - [hemat_transcriptome_v1.6.fasta.fai](https://gannet.fish.washington.edu/Atumefaciens/20210308_hemat_trinity_v1.6_v1.7/hemat_transcriptome_v1.6.fasta_trinity_out_dir/hemat_transcriptome_v1.6.fasta.fai)

- [Notebook entry](https://robertslab.github.io/sams-notebook/2021/03/08/Transcriptome-Assembly-Hematodinium-Transcriptomes-v1.6-and-v1.7-with-Trinity-on-Mox.html)

- BUSCOs: `C:26.5%[S:20.7%,D:5.8%],F:11.2%,M:62.3%,n:978`

  - [Notebook entry](https://robertslab.github.io/sams-notebook/2020/08/14/Transcriptome-Assessment-BUSCO-Metazoa-on-Hematodinium-v1.6-v1.7-v2.1-and-v3.1-on-Mox.html)

- BLASTx Annotation

  - [hemat_transcriptome_v1.6.fasta.blastx.outfmt6](https://gannet.fish.washington.edu/Atumefaciens/20200814_hemat_diamond_blastx_v1.6_v1.7_v2.1_v3.1/hemat_transcriptome_v1.6.fasta.blastx.outfmt6)

  - [Notebook entry](https://robertslab.github.io/sams-notebook/2020/08/14/Transcriptome-Annotation-Hematodinium-Transcriptomes-v1.6-v1.7-v2.1-v3.1-with-DIAMOND-BLASTx-on-Mox.html)

- GO Terms Annotation

  - [20210309.hemat_transcriptome_v1.6.fasta.trinotate.go_annotations.txt](https://gannet.fish.washington.edu/Atumefaciens/20210309_hemat_trinotate_transcriptome-v1.6/20210309.hemat_transcriptome_v1.6.fasta.trinotate.go_annotations.txt) (Trinotate)

  - [Notebook entry](https://robertslab.github.io/sams-notebook/posts/2021/2021-03-09-Transcriptome-Annotation---Trinotate-Hematodinium-v1.6-on-Mox/index.html)
</code></pre></div>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/hemat_transcriptome_v1.5.fasta">hemat_transcriptome_v1.5.fasta</a></p>
<ul>
<li>
<p>MD5 = <code>b8d4a3c1bad2e07da8431bf70bdabfdd</code></p>
</li>
<li>
<p><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-03-31-Transcriptome-Assessment---BUSCO-Metazoa-on-Hematodinium-MEGAN-Transcriptome/">BUSCOs</a>: <code>C:25.6%[S:20.7%,D:4.9%],F:11.7%,M:62.7%,n:978</code></p>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>)</p>
<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/hemat_transcriptome_v1.5.fasta.fai">hemat_transcriptome_v1.5.fasta.fai</a> : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/hemat_transcriptome_v1.5.fasta.fai</code></li>
</ul>
</li>
<li>
<p>Updated assembly from <a href="https://robertslab.github.io/sams-notebook/2020/03/30/Transcriptome-Assembly-Hematodinium-with-MEGAN6-Taxonomy-specific-Reads-with-Trinity-on-Mox.html">20200330</a>.</p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200331_hemat_diamond_blastx_megan/20200408.hemat.megan.Trinity.blastx.outfmt6">BLASTx Annotation</a> (txt; 355KB)</p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200408_hemat_trinotate_megan/20200408.hemat.trinotate.go_annotations.txt">Trinotate GO Terms Annotation</a> (txt; 2.3MB)</p>
</li>
<li>
<p>internal short-hand: includes 2018, 2019, 2020-GW with <em>Alveolata</em> only reads.</p>
</li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/hemat_transcriptome_v1.0.fasta">hemat_transcriptome_v1.0.fasta</a> (3.9MB)</p>
<ul>
<li>
<p>MD5 = <code>fa5eb74767d180af5265d2d1f80b6430</code></p>
</li>
<li>
<p><a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-02-07-Transcriptome-Assessment---BUSCO-Metazoa-on-Hematodinium-MEGAN-Transcriptome/index.html">BUSCOs</a>: <code>C:25.1%[S:19.2%,D:5.9%],F:9.5%,M:65.4%,n:978</code></p>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>)</p>
<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/hemat_transcriptome_v1.0.fasta.fai">hemat_transcriptome_v1.0.fasta.fai</a> : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/hemat_transcriptome_v1.0.fasta.fai</code></li>
</ul>
</li>
<li>
<p>Initial Trinity assembly from <a href="https://robertslab.github.io/sams-notebook/posts/2020/2020-01-22-Transcriptome-Assembly---Hematodinium-with-MEGAN6-Taxonomy-specific-Reads-with-Trinity-on-Mox/index.html">20200122</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200123_hemat_diamond_blastx_megan/20200122.hemat.megan.Trinity.blastx.outfmt6">BLASTx Annotation</a> (txt; 308KB)</p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20200126_hemat_trinotate_megan/20200126.hemat.trinotate.go_annotations.txt">Trinotate GO Terms Annotation</a> (txt; 2.1MB)</p>
</li>
<li>
<p>internal short-hand: includes 2018, 2019 with <em>Alveolata</em> only reads.</p>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="metacarcinus-magister-cancer-magister"><em>Metacarcinus magister</em> (<em>Cancer magister</em>)</h2>
<p>Genome:</p>
<ul>
<li>
<p><code>mmag_pilon_scaffolds.fasta</code></p>
<ul>
<li>
<p>MD5 = 5dfa2ba11edf0ff8191f112e0b1378d1</p>
</li>
<li>
<p>Not shared publicly until permission received from NOAA.</p>
</li>
<li>
<p>Roberts Lab members can access on Owl: <code>/web/halfshell/genomic-databank/mmag_pilon_scaffolds.fasta</code></p>
</li>
<li>
<p>Original filename: <code>pilon_scaffolds.fasta</code></p>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>)</p>
<ul>
<li><code>mmag_pilon_scaffolds.fasta.fai</code></li>
</ul>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="montipora-capitata"><em>Montipora capitata</em></h2>
<p>Genomes:</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCA_006542545.1_Mcap_UHH_1.1_genomic.fna"><code>GCA_006542545.1_Mcap_UHH_1.1_genomic.fna</code></a> (569MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>25efbc3110c0791b5eb2e5ac5c2a472f</code></p>
</li>
<li>
<p>Downloaded 20230125: <a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_006542545.1/">https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_006542545.1/</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Montipora_capitata_HIv3.assembly.fasta"><code>Montipora_capitata_HIv3.assembly.fasta</code></a> (745MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>99819eadba1b13ed569bb902eef8da08</code></p>
</li>
<li>
<p>Downloaded 2023017: <a href="http://cyanophora.rutgers.edu/montipora/">http://cyanophora.rutgers.edu/montipora/</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Indexes (<a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>)</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230131-mcap-HIv3-hisat2-build-index/Montipora_capitata_HIv3-hisat2-indices.tar.gz"><code>Montipora_capitata_HIv3-hisat2-indices.tar.gz</code></a> (tarball gzip; 1.2GB)</p>
<ul>
<li>
<p>MD5 checksum: <code>c8accb6c54e843198c776f0d6f0c603d</code></p>
</li>
<li>
<p>Needs to be unpacked before use!</p>
</li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Montipora_capitata_HIv3.genes.gff3"><code>Montipora_capitata_HIv3.genes.gff3</code></a> (67MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>5f6b80ba2885471c8c1534932ccb7e84</code></p>
</li>
<li>
<p>Downloaded 2023017: <a href="http://cyanophora.rutgers.edu/montipora/">http://cyanophora.rutgers.edu/montipora/</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230127-mcap-gff_to_gtf/Montipora_capitata_HIv3.genes.gtf"><code>Montipora_capitata_HIv3.genes.gtf</code></a> (101MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>ceef8eca945199415b23d2f1f0dd2066</code></p>
</li>
<li>
<p>Created 2023017: <a href="https://robertslab.github.io/sams-notebook/2023/01/27/Data-Wrangling-M.capitata-Genome-GFF-to-GTF-Using-gffread.html">https://robertslab.github.io/sams-notebook/2023/01/27/Data-Wrangling-M.capitata-Genome-GFF-to-GTF-Using-gffread.html</a></p>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="mytilus-trossulus"><em>Mytilus trossulus</em></h2>
<p>Genome:</p>
<ul>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/assembly/GCF_021869535.1/">NCBI Assembly GCF_021869535.1</a></p>
</li>
<li>
<p><a href="https://doi.org/10.1016/j.margen.2021.100906">A chromosome-level reference genome of the bay mussel Mytilus trossulus</a></p>
</li>
<li>
<p><a href="https://www.ncbi.nlm.nih.gov/genome/annotation_euk/Mytilus_trossulus/100/">NCBI Mytilus trossulus Annotation Release 100</a></p>
</li>
</ul>
<p>Transcriptome:</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Mtros-hq_transcripts.fasta">Mtros-hq_transcripts.fasta</a></p>
<ul>
<li>MD5 = 381f7b6970fd20ff6b0e72006c80a</li>
</ul>
</li>
</ul>
<h2 id="ostrea-lurida"><em>Ostrea lurida</em></h2>
<p>Genome:</p>
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081.fa">Olurida_v081.fa</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081.fa</code></p>
<ul>
<li>
<p>MD5 = 3ac56372bd62038f264d27eef0883bd3</p>
</li>
<li>
<p>This is <code>v080</code> with only contigs &gt; 1000bp. Details of how <code>v080</code> was reduced found <a href="https://github.com/sr320/nb-2018/blob/master/O_lurida/04-Pbjelly-10k-mapping.ipynb">here</a>.</p>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>)</p>
<ul>
<li><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081.fa.fai">Olurida_v081.fa.fai</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081.fa.fai</code></li>
</ul>
</li>
</ul>
</li>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v080.fa">Olurida_v080.fa</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v080.fa</code></p>
<ul>
<li>
<p>MD5 = 9258398f554493e08fdc30e9c1409864</p>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>)</p>
<ul>
<li><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v080.fa.fai">Olurida_v080.fa.fai</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v080.fa.fai</code></li>
</ul>
</li>
<li>
<p>Also known as <code>pbjelly_sjw_01</code>. Details found <a href="https://github.com/sr320/LabDocs/blob/master/jupyter_nbs/sam/20171130_emu_pbjelly.ipynb">here</a>, though confirmation would be good.</p>
</li>
</ul>
</li>
</ul>
<p>Bisulfite Genomes:</p>
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Crassostrea_gigas.oyster_v9.dna_sm.toplevel_bisulfite.tar.gz">Olurida_v080_bisulfite.tar.gz</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v080_bisulfite.tar.gz</code></p>
</li>
<li>
<p>Gzipped tarball of bisulfite genome for use with Bismark</p>
</li>
<li>
<p>Creation details <a href="https://robertslab.github.io/sams-notebook/posts/2018/2018-05-08-bs-seq-mapping-olympia-oyster-bisulfite-sequencing-trimgalore-fastqc-bismark/index.html">here</a></p>
</li>
</ul>
<p>Transcriptomes:</p>
<ul>
<li>
<p><a href="http://eagle.fish.washington.edu/cnidarian/Olurida_transcriptome_v3.fasta">Olurida_transcriptome_v3.fasta</a></p>
<ul>
<li>MD5 = 9da3242af2be0463051ec7e1f39b2593</li>
</ul>
</li>
</ul>
<p><strong>Tissue-specific transcriptomes generated by Katherine Silliman</strong></p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_CA_adductor_Trinity.fasta.gz">Olurida_CA_adductor_Trinity.fasta.gz</a></p>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_CA_ctenidia_Trinity.fasta.gz">Olurida_CA_ctenidia_Trinity.fasta.gz</a></p>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_CA_mantle_Trinity.fasta.gz">Olurida_CA_mantle_Trinity.fasta.gz</a></p>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_gonad_Trinity.fasta.gz">Olurida_gonad_Trinity.fasta.gz</a></p>
</li>
</ul>
<p><strong>Genome Feature Tracks</strong></p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081_genome_snap02.all.renamed.putative_function.domain_added.gff">Olurida_v081_genome_snap02.all.renamed.putative_function.domain_added.gff</a> (2.9GB) : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081_genome_snap02.all.renamed.putative_function.domain_added.gff</code></p>
<ul>
<li>MD5 = <code>f54512bd964f45645c34b1e8e403a2b0</code></li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.CDS.gff">Olurida_v081-20190709.CDS.gff</a> : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.CDS.gff</code></p>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.exon.gff">Olurida_v081-20190709.exon.gff</a> : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.exon.gff</code></p>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.gene.gff">Olurida_v081-20190709.gene.gff</a> : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.gene.gff</code></p>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.mRNA.gff">Olurida_v081-20190709.mRNA.gff</a> : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081-20190709.mRNA.gff</code></p>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081_TE-Cg.gff">Olurida_v081_TE-Cg.gff</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081_TE-Cg.gff</code></p>
<ul>
<li>
<p>MD5 = 977fd7cdb460cd0b9df5e875e1e880ea</p>
</li>
<li>
<p>Transposable Element track - more details in <a href="https://robertslab.github.io/sams-notebook/posts/2018/2018-07-03-transposable-element-mapping-olympia-oyster-genome-assembly-olurida_v081-using-repeatmasker-4-07/index.html">Sam's Notebook</a>, including a <a href="http://owl.fish.washington.edu/Athaliana/20180702_oly_repeatmasker_Cgigas/Olurida_v081.fa.tbl">summary table</a>.</p>
</li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081_CG-motif.gff">Olurida_v081_CG-motif.gff</a> : <code>https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081_CG-motif.gff</code></p>
</li>
</ul>
<hr />
<h2 id="ostrea-chil"><em>Ostrea chil</em></h2>
<p>Genome:</p>
<ul>
<li>
<p>[HapA]</p>
</li>
<li>
<p>[HapB]</p>
</li>
</ul>
<p>Protein:</p>
<p>-</p>
<p>Blast:</p>
<h2 id="panopea-generosa"><em>Panopea generosa</em></h2>
<p>Genome:</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.fa">Panopea-generosa-v1.0.fa</a> : <code>https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.fa</code></p>
<ul>
<li>
<p><a href="https://www.ebi.ac.uk/ena/data/view/GCA_902825435">ENA Accession: GCA_902825435</a></p>
</li>
<li>
<p>Version of 070 containing 18 largest scaffolds (<a href="https://robertslab.github.io/sams-notebook/posts/2019/2019-06-25-Data-Wrangling---FastA-Subsetting-of-Pgenerosa_v070.fa-Using-samtools-faidx/index.html">details on subsetting</a>)</p>
</li>
<li>
<p><a href="https://robertslab.github.io/sams-notebook/2019/11/05/Data-Wrangling-Rename-Pgenerosa_v074-Files-and-Scaffolds.html">FastA file and scaffolds were renamed on 20191105</a> (notebook)</p>
</li>
<li>
<p>MD5 = 32976550b9030126c07920d5f2db179c</p>
</li>
<li>
<p>BUSCO scores:</p>
<ul>
<li>
<p><code>C:71.6%[S:70.7%,D:0.9%],F:4.7%,M:23.7%,n:978</code></p>
</li>
<li>
<p><a href="https://robertslab.github.io/sams-notebook/posts/2019/2019-07-10-Genome-Assessment---BUSCO-Metazoa-on-Pgenerosa_v074-on-Mox/">Notebook entry</a></p>
</li>
</ul>
</li>
<li>
<p>FastA index (<code>samtools faidx</code>):</p>
<ul>
<li><code>https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.fa.fai</code></li>
</ul>
</li>
<li>
<p>Gene annotation file:</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220419-pgen-gene_annotation_mapping/20220419-pgen-gene-accessions-gene_id-gene_name-gene_description-alt_gene_description-go_ids.tab">20220419-pgen-gene-accessions-gene_id-gene_name-gene_description-alt_gene_description-go_ids.tab</a></p>
<ul>
<li>
<p><code>gene_ID</code>: Gene ID from our <a href="http://en.wikipedia.org/wiki/Geoduck"><em>Panopea generosa</em> (Pacific geoduck)</a> genome.</p>
</li>
<li>
<p><code>SPIDs</code>: Semicolon-delimited list of SPIDs from UniProt. One SPID in this list is a match corresponding to the our original BLAST annotations.</p>
</li>
<li>
<p><code>UniProt_gene_ID</code>: Gene accession from UniProt.</p>
</li>
<li>
<p><code>gene</code>: Abbreviated gene name from UniProt.</p>
</li>
<li>
<p><code>gene_description</code>: Human-readable gene description from UniProt.</p>
</li>
<li>
<p><code>alternate_gene_description</code>: Human-readable alternate gene description from UniProt.</p>
</li>
<li>
<p><code>GO_IDs</code>: Semicolon-delimited GO IDs from UniProt.</p>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>Bisulfite Genome:</p>
<p>Genome Feature Tracks:</p>
<ul>
<li>
<p>Panopea-generosa-vv0.74.a4</p>
<p><em>These originate from <a href="https://robertslab.github.io/sams-notebook/posts/2019/2019-09-28-Genome-Annotation---Pgenerosa_v074-a4-Using-GenSAS/index.html">GenSAS annotation on 20190928</a></em></p>
<p>Individual feature GFFs were made with the following shell commands:</p>
</li>
</ul>
<!-- -->

<div class="highlight"><pre><span></span><code>```bash

features_array=(CDS exon gene mRNA repeat_region rRNA tRNA)

input=&quot;Panopea-generosa-vv0.74.a4-merged-2019-10-07-4-46-46.gff3&quot;

for feature in ${features_array[@]}
  do
  output=&quot;Panopea-generosa-vv0.74.a4.${feature}.gff3&quot;
  head -n 3 ${input} \
  &gt;&gt; ${output}
  awk -v feature=&quot;$feature&quot; &#39;$3 == feature {print}&#39; ${input} \
  &gt;&gt; ${output}
done
```

- [GFF files and scaffolds were renamed on 20191105](https://robertslab.github.io/sams-notebook/2019/11/05/Data-Wrangling-Rename-Pgenerosa_v074-Files-and-Scaffolds.html) (notebook)
</code></pre></div>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.gff3">Panopea-generosa-v1.0.a4.gff3</a></p>
<ul>
<li>Primary GFF containing all features.</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220323-pgen-nextflow_rnaseq-tissues/genome/Panopea-generosa-v1.0.a4_biotype-trna_strand_converted-no_RNAmmer.gtf">Panopea-generosa-v1.0.a4_biotype-trna_strand_converted-no_RNAmmer.gtf</a></p>
<ul>
<li>
<p>GTF file with formatting to avoid downstream parsing problems.</p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/resources/issues/1411">GitHub Issue describing creation and problems</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.CpG.gff">Panopea-generosa-v1.0.CpG.gff</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.CDS.gff3">Panopea-generosa-v1.0.a4.CDS.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.exon.gff3">Panopea-generosa-v1.0.a4.exon.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.gene.gff3">Panopea-generosa-v1.0.a4.gene.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.intergenic.bed">Panopea-generosa-v1.0.a4.intergenic.bed</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.introns.bed">Panopea-generosa-v1.0.a4.introns.bed</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.mRNA.gff3">Panopea-generosa-v1.0.a4.mRNA.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.rRNA.gff3">Panopea-generosa-v1.0.a4.rRNA.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeat_region.gff3">Panopea-generosa-v1.0.a4.repeat_region.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeats.DNA.gff3">Panopea-generosa-v1.0.a4.repeats.DNA.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeats.LINE.gff3">Panopea-generosa-v1.0.a4.repeats.LINE.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeats.LTR.gff3">Panopea-generosa-v1.0.a4.repeats.LTR.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeats.RC.gff3">Panopea-generosa-v1.0.a4.repeats.RC.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeats.SINE.gff3">Panopea-generosa-v1.0.a4.repeats.SINE.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeats.Simple_repeat.gff3">Panopea-generosa-v1.0.a4.repeats.Simple_repeat.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.repeats.Unknown.gff3">Panopea-generosa-v1.0.a4.repeats.Unknown.gff3</a></p>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20191105_swoose_pgen_v074_renaming/Panopea-generosa-v1.0.a4.tRNA.gff3">Panopea-generosa-v1.0.a4.tRNA.gff3</a></p>
</li>
</ul>
<hr />
<p>Fasta files:</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220324-pgen-gffs_to_fastas/Panopea-generosa-v1.0.a4.CDS.fasta">Panopea-generosa-v1.0.a4.CDS.fasta</a> (67M)</p>
<ul>
<li>MD5: <code>fb192eab0aefd5d3ba5bebef2a012f15</code></li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220324-pgen-gffs_to_fastas/Panopea-generosa-v1.0.a4.CDS.fasta.fai">Panopea-generosa-v1.0.a4.CDS.fasta.fai</a> (26M)</p>
<ul>
<li>MD5: <code>f2266a449290ea0383d2eb98eb3ed426</code></li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220324-pgen-gffs_to_fastas/Panopea-generosa-v1.0.a4.gene.fasta">Panopea-generosa-v1.0.a4.gene.fasta</a> (362M)</p>
<ul>
<li>MD5: <code>7c956b1c27d14bd91959763403f81265 588d18f5fe0e4f2259a25586349fc244</code></li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220324-pgen-gffs_to_fastas/Panopea-generosa-v1.0.a4.gene.fasta.fai">Panopea-generosa-v1.0.a4.gene.fasta.fai</a> (2.4M)</p>
<ul>
<li>MD5: <code>588d18f5fe0e4f2259a25586349fc244</code></li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220324-pgen-gffs_to_fastas/Panopea-generosa-v1.0.a4.mRNA.fasta">Panopea-generosa-v1.0.a4.mRNA.fasta</a> (475M)</p>
<ul>
<li>MD5: <code>1823be75694cf70f0ea6f1abc072ba16 e120b4c1d3bb0917868e72cd22507bbc</code></li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220324-pgen-gffs_to_fastas/Panopea-generosa-v1.0.a4.mRNA.fasta.fai">Panopea-generosa-v1.0.a4.mRNA.fasta.fai</a> (3.4M)</p>
<ul>
<li>MD5: <code>e120b4c1d3bb0917868e72cd22507bbc</code></li>
</ul>
</li>
</ul>
<p>Jupyter notebook with creation deets (NB Viewer):</p>
<ul>
<li><a href="https://nbviewer.org/github/RobertsLab/code/blob/master/notebooks/sam/20220324-pgen-gffs_to_fastas.ipynb">20220324-pgen-gffs_to_fastas.ipynb</a></li>
</ul>
<p>CDS FastA description lines look like this:</p>
<ul>
<li><code>PGEN_.00g000010.m01.CDS01|PGEN_.00g000010.m01::Scaffold_01:2-125</code></li>
</ul>
<p>Explanation for CDS:</p>
<ul>
<li><code>PGEN_.00g000010.m01.CDS01</code>: Unique sequence ID.</li>
<li><code>PGEN_.00g000010.m01</code>: "Parent" ID. Corresponds to unique <em>mRNA</em> ID.</li>
<li><code>Scaffold_01</code>: Originating scaffold.</li>
<li><code>2-125</code>: Sequence coordinates from scaffold mentioned above.</li>
</ul>
<p>mRNA FastA description looks like this:</p>
<ul>
<li><code>PGEN_.00g000030.m01|PGEN_.00g000030::Scaffold_01:49248-52578</code></li>
</ul>
<p>Explanation for mRNA:</p>
<ul>
<li><code>PGEN_.00g000030.m01</code>: Unique sequence ID.</li>
<li><code>PGEN_.00g000030</code>: "Parent" ID. Corresponds to unique <em>gene</em> ID.</li>
<li><code>Scaffold_01</code>: Originating scaffold.</li>
<li><code>49248-52578</code>: Sequence coordinates from scaffold mentioned above.</li>
</ul>
<hr />
<ul>
<li>
<p><a href="http://owl.fish.washington.edu/halfshell/genomic-databank/Pgenerosa_transcriptome_v5.fasta">Pgenerosa_transcriptome_v5.fasta</a> : <code>http://owl.fish.washington.edu/halfshell/genomic-databank/Pgenerosa_transcriptome_v5.fasta</code></p>
<ul>
<li>MD5 = <code>5a21424ecbc88c3b01daefe56bed79da</code></li>
</ul>
</li>
</ul>
<p>Transcriptome generated from various libaries - details <a href="https://robertslab.github.io/sams-notebook/2018/09/04/transcriptome-assembly-geoduck-rnaseq-data.html">here</a>.</p>
<ul>
<li>
<p>Kallisto index for <code>Pgenerosa_transcriptome_v5.fasta</code> (8.2GB):</p>
<ul>
<li><a href="https://gannet.fish.washington.edu/seashell/wd/062821/transcriptome_v5.idx"><code>https://gannet.fish.washington.edu/seashell/wd/062821/transcriptome_v5.idx</code></a></li>
</ul>
</li>
</ul>
<hr />
<h2 id="pocillipora-acuta"><em>Pocillipora acuta</em></h2>
<p>Genome:</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_acuta_HIv2.assembly.fasta"><code>Pocillopora_acuta_HIv2.assembly.fasta</code></a> (389MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>ce3b69ff3f5dafb8fb7416dc862ef4a0</code></p>
</li>
<li>
<p>Downloaded 20230125: <a href="http://cyanophora.rutgers.edu/Pocillopora_acuta/">http://cyanophora.rutgers.edu/Pocillopora_acuta/</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Index (<a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>):</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230131-pacu-HIv2-hisat2-build-index/Pocillopora_acuta_HIv2-hisat2-indices.tar.gz"><code>Pocillopora_acuta_HIv2-hisat2-indices.tar.gz</code></a> (597MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>80dbf8ca589f569f43ef2a75ab57e17d</code></p>
</li>
<li>
<p>Needs to be unpacked before use!</p>
</li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks:</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_acuta_HIv2.genes.gff3"><code>Pocillopora_acuta_HIv2.genes.gff3</code></a> (54MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>fad5aa85afd7e3bec4400ca6da7d706d</code></p>
</li>
<li>
<p>Downloaded 20230125: <a href="http://cyanophora.rutgers.edu/Pocillopora_acuta/">http://cyanophora.rutgers.edu/Pocillopora_acuta/</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230126-pacu-gff_to_gtf/Pocillopora_acuta_HIv2.gtf"><code>Pocillopora_acuta_HIv2.gtf</code></a> (82MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>34196bd945eb4965e665097648037132</code></p>
</li>
<li>
<p>Created 20230127: <a href="https://robertslab.github.io/sams-notebook/posts/2023/2023-01-26-Data-Wrangling---P.acuta-Genome-GFF-to-GTF-Conversion-Using-gffread/index.html">Data-Wrangling-P.acuta-Genome-GFF-to-GTF-Conversion-Using-gffread.html</a></p>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="pocillopora-meandrina"><em>Pocillopora meandrina</em></h2>
<p>Genome(s):</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.assembly.fasta"><code>Pocillopora_meandrina_HIv1.assembly.fasta</code></a> (360MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>36eb9cdaf92db69906e6d1486a8406f5</code></p>
</li>
<li>
<p>Downloaded 20230519: <a href="http://cyanophora.rutgers.edu/Pocillopora_meandrina/">http://cyanophora.rutgers.edu/Pocillopora_meandrina/</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Indexes (<a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>):</p>
<ul>
<li>
<p><a href=""><code>Pocillopora_meandrina_HIv1.assembly-hisat2-indices.tar.gz</code></a> (tarball gzip; MB)</p>
<ul>
<li>
<p>MD5 checksum: ``</p>
</li>
<li>
<p>Needs to be unpacked before use!</p>
</li>
<li>
<p>Notebook: <a href=""></a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks</p>
<ul>
<li>
<p>Genes</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.genes-valid.gff3">https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.genes-valid.gff3</a> (55MB)</p>
</li>
<li>
<p>MD5 checksum: <code>5865589d1f2764b4b74df91ea78d5652</code></p>
</li>
<li>
<p>A GFF3-compliant version of <code>Pocillopora_meandrina_HIv1.genes.gff3</code> (see below). Created GFF3 compliant version via the following command (replace <code>transcript</code> with <code>mRNA</code>):</p>
<ul>
<li><code>sed 's/transcript/mRNA/' Pocillopora_meandrina_HIv1.genes.gff3 &gt; Pocillopora_meandrina_HIv1.genes-valid.gff3</code></li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pocillopora_meandrina_HIv1.genes.gff3">Pocillopora_meandrina_HIv1.genes.gff3</a> (55MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>ace5c9a588321fada8e6771a1c758861</code></p>
</li>
<li>
<p>Downloaded 20230519: <a href="http://cyanophora.rutgers.edu/Pocillopora_meandrina/">http://cyanophora.rutgers.edu/Pocillopora_meandrina/</a></p>
</li>
<li>
<p>NOTE: This is labelled as a GFF3, but in reality closer to a GTF file; as it only contains transcript/exon/CDS features. There are no gene features (e.g. 5'/3'UTR, mRNA, gene, etc). Additionally, the feature label of <code>transcript</code> is not GFF or GTF compliant.</p>
</li>
</ul>
</li>
</ul>
</li>
<li>
<p>Repeats</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230526-pmea-repeatmasker-Pocillopora_meandrina_HIv1.assembly/Pocillopora_meandrina_HIv1.assembly.fasta.out.gff">Pocillopora_meandrina_HIv1.assembly.fasta.out.gff</a> (13MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>6e7a25bf51a7c838b9659dd7ec37990f</code></p>
</li>
<li>
<p>Notebook: <a href="https://robertslab.github.io/sams-notebook/posts/2023/2023-05-26-Repeats-Identification---P.meandrina-Using-RepeatMasker-on-Mox/index.html">Repeats-Identification-P.meandrina-Using-RepeatMasker-on-Mox.html</a></p>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<hr />
<h2 id="pocillipora-verrucosa"><em>Pocillipora verrucosa</em></h2>
<p>Genomes:</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCA_014529365.1_Pver_genome_assembly_v1.0_genomic.fna"><code>GCA_014529365.1_Pver_genome_assembly_v1.0_genomic.fna</code></a> (369MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>6ca98fae6a8b86183d75b23cf52a6651</code></p>
</li>
<li>
<p>Downloaded 20230125: <a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_014529365.1/">https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_014529365.1/</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pver_genome_assembly_v1.0.fasta"><code>Pver_genome_assembly_v1.0.fasta</code></a> (363MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>cb1ed5a1b724d92456347a28bb25f228</code></p>
</li>
<li>
<p>Downloaded 20230127: <a href="http://pver.reefgenomics.org/download/">http://pver.reefgenomics.org/download/</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Indexes (<a href="https://daehwankimlab.github.io/hisat2/"><code>HISAT2</code></a>):</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230125-pver-hisat2-GCA_014529365.1-index/pver-GCA_014529365.1-hisat2-indices.tar.gz"><code>pver-GCA_014529365.1-hisat2-indices.tar.gz</code></a> (tarball gzip; 563MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>f1669e7d88cf014fcfa10c6c06e03802</code></p>
</li>
<li>
<p>Needs to be unpacked before use!</p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230131-pver-v1.0-hisat2-build-index/Pver_genome_assembly_v1.0-hisat2-indices.tar.gz"><code>Pver_genome_assembly_v1.0-hisat2-indices.tar.gz</code></a> (tarball gzip; 594MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>57e193e101396fab67de04c851f63240</code></p>
</li>
<li>
<p>Needs to be unpacked before use!</p>
</li>
<li>
<p>Notebook: <a href="https://robertslab.github.io/sams-notebook/posts/2023/2023-01-31-Genome-Indexing---P.verrucosa-v1.0-Assembly-with-HiSat2-on-Mox/index.html">Genome-Indexing-P.verrucosa-v1.0-Assembly-with-HiSat2-on-Mox.html</a></p>
</li>
</ul>
</li>
</ul>
<p>Genome Feature Tracks</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pver_genome_assembly_v1.0.gff3">~~<code>Pver_genome_assembly_v1.0.gff3</code>~~</a> (70MB)</p>
<ul>
<li>
<p><strong>NOTE: DO <em>NOT</em> USE! NOT A VALID GFF3 FORMAT!</strong></p>
<ul>
<li>
<p>Retaining to maintain provenance of data.</p>
</li>
<li>
<p>Use updated/validated <a href="https://gannet.fish.washington.edu/Atumefaciens/20230127-pver-gff_to_gtf/Pver_genome_assembly_v1.0-valid.gff3"><code>Pver_genome_assembly_v1.0-valid.gff3</code></a>.</p>
</li>
</ul>
</li>
<li>
<p>MD5 checksum: <code>3f1d52afa2801f9aa126623aba3c149d</code></p>
</li>
<li>
<p>Downloaded 20230127: <a href="http://pver.reefgenomics.org/download/">http://pver.reefgenomics.org/download/</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230127-pver-gff_to_gtf/Pver_genome_assembly_v1.0-valid.gff3"><code>Pver_genome_assembly_v1.0-valid.gff3</code></a> (70MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>5dd8f21a4faea1f46c48a5ab253749d7</code></p>
</li>
<li>
<p>Modified/validated version of <code>Pver_genome_assembly_v1.0.gff3</code></p>
</li>
<li>
<p>Created 20230127: <a href="https://robertslab.github.io/sams-notebook/posts/2023/2023-01-27-Data-Wrangling---P.verrucosa-Genome-GFF-to-GTF-Using-gffread/index.html">Data-Wrangling-P.verrucosa-Genome-GFF-to-GTF-Using-gffread</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230127-pver-gff_to_gtf/Pver_genome_assembly_v1.0-valid.gtf"><code>Pver_genome_assembly_v1.0-valid.gtf</code></a> (48MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>c3cc8fb576bcf39dd17b6d229100aa56</code></p>
</li>
<li>
<p>Created 20230127: <a href="https://robertslab.github.io/sams-notebook/posts/2023/2023-01-27-Data-Wrangling---P.verrucosa-Genome-GFF-to-GTF-Using-gffread/index.html">Data-Wrangling-P.verrucosa-Genome-GFF-to-GTF-Using-gffread</a></p>
</li>
</ul>
</li>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20230127-pver-gff_to_gtf/Pver_genome_assembly_v1.0-valid.genes.bed"><code>Pver_genome_assembly_v1.0-valid.genes.bed</code></a> (1.2MB)</p>
<ul>
<li>
<p>MD5 checksum: <code>f19672f65c4e376f96c6ab23c202c2e0</code></p>
</li>
<li>
<p>Created 20230227 by Sam White:</p>
<div class="highlight"><pre><span></span><code>bedops_linux_x86_64-v2.4.40/gff2bed<span class="w"> </span><span class="se">\</span>
&lt;<span class="w"> </span>Pver_genome_assembly_v1.0-valid.genes.gff3<span class="w"> </span><span class="se">\</span>
<span class="p">|</span><span class="w"> </span>awk<span class="w"> </span>-F<span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;BEGIN {OFS=&quot;\t&quot;} {print $1, $2, $3, $4, $5, $6}&#39;</span><span class="w"> </span><span class="se">\</span>
&gt;<span class="w"> </span>Pver_genome_assembly_v1.0-valid.genes.bed
</code></pre></div>
</li>
</ul>
</li>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/Pver_CGmotif.gff">Pver_CGmotif.gff</a>: <code>https://owl.fish.washington.edu/halfshell/genomic-databank/Pver_CGmotif.gff</code> (1.3G)</p>
</li>
</ul>
<hr />
<h2 id="pycnopodia-helianthodes"><em>Pycnopodia helianthodes</em></h2>
<p>Genome</p>
<ul>
<li>
<p><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/GCA_032158295.1_ASM3215829v1_genomic.fna">GCA_032158295.1_ASM3215829v1_genomic.fna</a></p>
</li>
<li>
<p>MD5: <code>c6eb5b44d2bc14c37c852688a39009ad</code></p>
</li>
<li>
<p>Downloaded from NCBI 20231025 by SJW.</p>
</li>
</ul>
<hr />
<h2 id="qpx"><em>QPX</em></h2>
<p>Genome:</p>
<ul>
<li><a href="http://eagle.fish.washington.edu/QPX_genome/QPX_v017.fasta">QPX_v017.fasta</a> : <code>http://eagle.fish.washington.edu/QPX_genome/QPX_v017.fasta</code></li>
</ul>
<p>CLC v5.1 Mismatch cost = 2; Perform scaffolding = Yes; Mapping mode = Map reads back to contigs (slow); Deletion cost = 3; Similarity fraction = 0.9; Length fraction = 0.8; Insertion cost = 3; Update contigs = Yes; Automatic word size = Yes; Minimum contig length = 10000; Automatic bubble size = Yes; input: filtered_QPX_DNA_GTGAAA_L001_R1 trimmed.</p>
<ul>
<li><a href="https://ndownloader.figshare.com/files/3085550">QPX_v017.fasta</a> : <code>https://ndownloader.figshare.com/files/3085550</code></li>
</ul>
<p>CLC v5.1 Mismatch cost = 2; Perform scaffolding = Yes; Mapping mode = Map reads back to contigs (slow); Deletion cost = 3; Similarity fraction = 0.9; Length fraction = 0.8; Insertion cost = 3; Update contigs = Yes; Automatic word size = Yes; Minimum contig length = 10000; Automatic bubble size = Yes; input: filtered_QPX_DNA_GTGAAA_L001_R1 trimmed.</p>
<ul>
<li><a href="https://doi.org/10.1371/journal.pone.0074196.s001">QPX_v015.fasta</a> : <code>https://doi.org/10.1371/journal.pone.0074196.s001</code></li>
</ul>
<blockquote>
<p>De novo assembly was performed with Genomics Workbench v. 5.0 (CLC Bio, Germany) on quality trimmed sequences with the following parameters: mismatch cost = 2, deletion cost = 3, similarity fraction = 0.9, insertion cost = 3, length fraction = 0.8 and minimum contig size of 100 bp for genomic data and 200 bp for transcriptomic data. In order to remove ribosomal RNA sequences from the transcriptome data, consensus sequences were compared to the NCBI nt database using the BLASTn algorithm [59]. Sequences with significant matches (9) were removed and not considered in subsequent analyses.</p>
</blockquote>
<p>Manuscript: <a href="https://doi.org/10.1371/journal.pone.0074196">https://doi.org/10.1371/journal.pone.0074196</a></p>
<p>Transcriptome:</p>
<ul>
<li><a href="http://eagle.fish.washington.edu/cnidarian/QPX_transcriptome_v1_clean.fasta">QPX_transcriptome_v1_clean.fasta</a></li>
</ul>
<p>QPX_Transcriptome v2.1</p>
<p>Subset of version 1 (v1) that only includes sequences with e-value \&lt; 1E-20. Based on Swiss-Prot blastx output, all sequences are oriented 5' - 3'. nucleotides between stop codons; minimum size 200.</p>
<h2 id="salvelinus-namaycush-lake-trout"><em>Salvelinus namaycush</em> (lake trout)</h2>
<p>Genome:</p>
<ul>
<li><a href="https://owl.fish.washington.edu/halfshell/genomic-databank/SaNama_1.0_genomic.fna">SaNama_1.0_genomic.fna</a>: <code>https://owl.fish.washington.edu/halfshell/genomic-databank/SaNama_1.0_genomic.fna</code></li>
</ul>
<p>Genome Feature Tracks:</p>
<ul>
<li>
<p><a href="https://gannet.fish.washington.edu/Atumefaciens/20220818-snam-gff_to_bed-genes/20220818-snam-GCF_016432855.1_SaNama_1.0_genes.bed">20220818-snam-GCF_016432855.1_SaNama_1.0_genes.bed</a></p>
<ul>
<li><a href="https://robertslab.github.io/sams-notebook/posts/2022/2022-08-18-Data-Wrangling---Convert-S.namaycush-NCBI-GFF-to-genes-only-BED-file-for-Use-in-Ballgown-Analysis/index.html">Notebook entry</a></li>
</ul>
</li>
</ul>
<hr />












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.action.edit", "content.code.copy"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>