
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../bio_Basics/">
      
      
        <link rel="next" href="../bio_DNA-methylation/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Annotation - Roberts Lab Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#annotation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Roberts Lab Handbook" class="md-header__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Roberts Lab Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Annotation
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6m0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4M7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Roberts Lab Handbook" class="md-nav__button md-logo" aria-label="Roberts Lab Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Roberts Lab Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/RobertsLab/resources/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Lab Culture
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Lab Culture
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Onboarding/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Onboarding
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Code-of-Conduct/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Code of Conduct
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Environment-and-Expectations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Environment and Expectations
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Communication/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Communication
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Notebooks/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Notebooks
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../DEI-Resources/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DEI Resources
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../External-Communication-and-Funding/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    External Communication
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Safety/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Safety
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Undergraduate-contract/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Undergraduate Contract
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../How-to-Grad-School-Slides/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How to Grad School (Outreach Slides)
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Computing and Code
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Computing and Code
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Best-Practices/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Best Practices
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Agentic-Coding-Tools/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Agentic Coding Tools
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../code_Snippets/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Code Snippets
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Computing-Hardware/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Computing Hardware
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Software/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Software
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_6" >
        
          
          <label class="md-nav__link" for="__nav_3_6" id="__nav_3_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Klone Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_6">
            <span class="md-nav__icon md-icon"></span>
            Klone Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_quick-start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quick Start
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_containers/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Containers
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Conda/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Conda
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Adding-a-User/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    New User
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Logging-In/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Logging in
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Node-Types/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Node Types
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Data-Storage-and-System-Organization/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data Storage and System Organization
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_File-Transfers/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    File Transfers
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Installing-Programs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Installing Programs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_Running-a-Job/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Running a job
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_RStudio-Server/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RStudio Server
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../klone_VS-Code/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    VS Code
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3_7" >
        
          
          <label class="md-nav__link" for="__nav_3_7" id="__nav_3_7_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Raven Guides
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_7">
            <span class="md-nav__icon md-icon"></span>
            Raven Guides
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../raven_Conda/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Conda
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../raven_RStudio-Server/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    RStudio Server
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Databases and Catalogs
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Databases and Catalogs
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Lab-Inventory/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lab Inventory
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Pubathon/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Pubathon
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Digital-Media/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Digital Media
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Experiment-Database/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Experiment Database
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Genomic-Resources/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Genomic Resources
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../music/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Spotify Playlists
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Protocols
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Purchasing-and-Reimbursement/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Purchasing and Reimbursement
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Data-Management/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data-Management
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5_3" >
        
          
          <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Lab Protocols
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5_3">
            <span class="md-nav__icon md-icon"></span>
            Lab Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/protocol-bullet_blender/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Bullet Blender
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/protocol-qPCR/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    qPCR
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/resazurin-assay/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Resazurin Assay
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../protocols/protocol-reverse_transcription/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Reverse Transcription
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Chemical-Standard-Operating-Protocols/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Chemical SOPs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Alarm-Information/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Alarm Information
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Bioinformatic Workflows
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            Bioinformatic Workflows
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Basics/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Basics
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Annotation
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Annotation
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#intro-blast" class="md-nav__link">
    <span class="md-ellipsis">
      Intro (Blast)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Intro (Blast)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast-notebooks" class="md-nav__link">
    <span class="md-ellipsis">
      Blast Notebooks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-ontology-go" class="md-nav__link">
    <span class="md-ellipsis">
      Gene Ontology (GO)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Gene Ontology (GO)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#retrieve-go-terms-from-uniprot-using-swissprot-ids" class="md-nav__link">
    <span class="md-ellipsis">
      Retrieve GO terms from UniProt Using SwissProt IDs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#map-go-ids-to-goslims" class="md-nav__link">
    <span class="md-ellipsis">
      Map GO IDs to GOslims
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enrichment-analysis-visualization-and-interpretation" class="md-nav__link">
    <span class="md-ellipsis">
      Enrichment Analysis Visualization and Interpretation
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Enrichment Analysis Visualization and Interpretation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#overview-from-results-to-story" class="md-nav__link">
    <span class="md-ellipsis">
      Overview: From Results to Story
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#best-practices-for-synthesizing-enrichment-results" class="md-nav__link">
    <span class="md-ellipsis">
      Best Practices for Synthesizing Enrichment Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Best Practices for Synthesizing Enrichment Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-prioritize-and-filter-results" class="md-nav__link">
    <span class="md-ellipsis">
      1. Prioritize and Filter Results
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-group-related-processes" class="md-nav__link">
    <span class="md-ellipsis">
      2. Group Related Processes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-compare-across-conditions" class="md-nav__link">
    <span class="md-ellipsis">
      3. Compare Across Conditions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r-packages-for-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      R Packages for Visualization
    </span>
  </a>
  
    <nav class="md-nav" aria-label="R Packages for Visualization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#essential-packages" class="md-nav__link">
    <span class="md-ellipsis">
      Essential Packages
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#working-with-your-enrichment-results" class="md-nav__link">
    <span class="md-ellipsis">
      Working with Your Enrichment Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Working with Your Enrichment Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#loading-your-results" class="md-nav__link">
    <span class="md-ellipsis">
      Loading Your Results
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplify-results-remove-redundancy" class="md-nav__link">
    <span class="md-ellipsis">
      Simplify Results (Remove Redundancy)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#creating-impactful-figures" class="md-nav__link">
    <span class="md-ellipsis">
      Creating Impactful Figures
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Creating Impactful Figures">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-dot-plot-most-common-and-effective" class="md-nav__link">
    <span class="md-ellipsis">
      1. Dot Plot (Most Common and Effective)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-bar-plot-clear-and-simple" class="md-nav__link">
    <span class="md-ellipsis">
      2. Bar Plot (Clear and Simple)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-networkenrichment-map-shows-relationships" class="md-nav__link">
    <span class="md-ellipsis">
      3. Network/Enrichment Map (Shows Relationships)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-upset-plot-for-multiple-gene-lists" class="md-nav__link">
    <span class="md-ellipsis">
      4. Upset Plot (For Multiple Gene Lists)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-heatmap-of-enriched-terms" class="md-nav__link">
    <span class="md-ellipsis">
      5. Heatmap of Enriched Terms
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-gene-concept-network" class="md-nav__link">
    <span class="md-ellipsis">
      6. Gene-Concept Network
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-visualization-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      Advanced Visualization Strategies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Advanced Visualization Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#combining-multiple-plots" class="md-nav__link">
    <span class="md-ellipsis">
      Combining Multiple Plots
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#custom-ordering-and-grouping" class="md-nav__link">
    <span class="md-ellipsis">
      Custom Ordering and Grouping
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adding-annotations-and-context" class="md-nav__link">
    <span class="md-ellipsis">
      Adding Annotations and Context
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#writing-a-compelling-discussion" class="md-nav__link">
    <span class="md-ellipsis">
      Writing a Compelling Discussion
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Writing a Compelling Discussion">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#structure-your-narrative" class="md-nav__link">
    <span class="md-ellipsis">
      Structure Your Narrative
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-discussion-paragraph" class="md-nav__link">
    <span class="md-ellipsis">
      Example Discussion Paragraph
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#complete-workflow-example" class="md-nav__link">
    <span class="md-ellipsis">
      Complete Workflow Example
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alternative-approaches-and-tools" class="md-nav__link">
    <span class="md-ellipsis">
      Alternative Approaches and Tools
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Alternative Approaches and Tools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-goplot-for-circular-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Using GOplot for Circular Visualization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#revigo-for-semantic-clustering" class="md-nav__link">
    <span class="md-ellipsis">
      REVIGO for Semantic Clustering
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shinygo-for-interactive-exploration" class="md-nav__link">
    <span class="md-ellipsis">
      ShinyGO for Interactive Exploration
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tips-for-publication-quality-figures" class="md-nav__link">
    <span class="md-ellipsis">
      Tips for Publication-Quality Figures
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#common-pitfalls-to-avoid" class="md-nav__link">
    <span class="md-ellipsis">
      Common Pitfalls to Avoid
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab" class="md-nav__link">
    <span class="md-ellipsis">
      Use Cases from Our Lab
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#additional-resources" class="md-nav__link">
    <span class="md-ellipsis">
      Additional Resources
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genome-features" class="md-nav__link">
    <span class="md-ellipsis">
      Genome features
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Genome features">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bedtoolsintersectbed" class="md-nav__link">
    <span class="md-ellipsis">
      bedtools::intersectbed
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#transcriptome-trinity" class="md-nav__link">
    <span class="md-ellipsis">
      Transcriptome (Trinity)
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio_DNA-methylation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    DNA Methylation Analysis
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio-Gene-expression/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Gene Expression
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bio_Transcriptome-assembly/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Transcriptome Assembly
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#intro-blast" class="md-nav__link">
    <span class="md-ellipsis">
      Intro (Blast)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Intro (Blast)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#blast-notebooks" class="md-nav__link">
    <span class="md-ellipsis">
      Blast Notebooks
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-ontology-go" class="md-nav__link">
    <span class="md-ellipsis">
      Gene Ontology (GO)
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Gene Ontology (GO)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#retrieve-go-terms-from-uniprot-using-swissprot-ids" class="md-nav__link">
    <span class="md-ellipsis">
      Retrieve GO terms from UniProt Using SwissProt IDs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#map-go-ids-to-goslims" class="md-nav__link">
    <span class="md-ellipsis">
      Map GO IDs to GOslims
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enrichment-analysis-visualization-and-interpretation" class="md-nav__link">
    <span class="md-ellipsis">
      Enrichment Analysis Visualization and Interpretation
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Enrichment Analysis Visualization and Interpretation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#overview-from-results-to-story" class="md-nav__link">
    <span class="md-ellipsis">
      Overview: From Results to Story
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#best-practices-for-synthesizing-enrichment-results" class="md-nav__link">
    <span class="md-ellipsis">
      Best Practices for Synthesizing Enrichment Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Best Practices for Synthesizing Enrichment Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-prioritize-and-filter-results" class="md-nav__link">
    <span class="md-ellipsis">
      1. Prioritize and Filter Results
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-group-related-processes" class="md-nav__link">
    <span class="md-ellipsis">
      2. Group Related Processes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-compare-across-conditions" class="md-nav__link">
    <span class="md-ellipsis">
      3. Compare Across Conditions
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#r-packages-for-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      R Packages for Visualization
    </span>
  </a>
  
    <nav class="md-nav" aria-label="R Packages for Visualization">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#essential-packages" class="md-nav__link">
    <span class="md-ellipsis">
      Essential Packages
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#working-with-your-enrichment-results" class="md-nav__link">
    <span class="md-ellipsis">
      Working with Your Enrichment Results
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Working with Your Enrichment Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#loading-your-results" class="md-nav__link">
    <span class="md-ellipsis">
      Loading Your Results
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplify-results-remove-redundancy" class="md-nav__link">
    <span class="md-ellipsis">
      Simplify Results (Remove Redundancy)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#creating-impactful-figures" class="md-nav__link">
    <span class="md-ellipsis">
      Creating Impactful Figures
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Creating Impactful Figures">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-dot-plot-most-common-and-effective" class="md-nav__link">
    <span class="md-ellipsis">
      1. Dot Plot (Most Common and Effective)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-bar-plot-clear-and-simple" class="md-nav__link">
    <span class="md-ellipsis">
      2. Bar Plot (Clear and Simple)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-networkenrichment-map-shows-relationships" class="md-nav__link">
    <span class="md-ellipsis">
      3. Network/Enrichment Map (Shows Relationships)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#4-upset-plot-for-multiple-gene-lists" class="md-nav__link">
    <span class="md-ellipsis">
      4. Upset Plot (For Multiple Gene Lists)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#5-heatmap-of-enriched-terms" class="md-nav__link">
    <span class="md-ellipsis">
      5. Heatmap of Enriched Terms
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#6-gene-concept-network" class="md-nav__link">
    <span class="md-ellipsis">
      6. Gene-Concept Network
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-visualization-strategies" class="md-nav__link">
    <span class="md-ellipsis">
      Advanced Visualization Strategies
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Advanced Visualization Strategies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#combining-multiple-plots" class="md-nav__link">
    <span class="md-ellipsis">
      Combining Multiple Plots
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#custom-ordering-and-grouping" class="md-nav__link">
    <span class="md-ellipsis">
      Custom Ordering and Grouping
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adding-annotations-and-context" class="md-nav__link">
    <span class="md-ellipsis">
      Adding Annotations and Context
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#writing-a-compelling-discussion" class="md-nav__link">
    <span class="md-ellipsis">
      Writing a Compelling Discussion
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Writing a Compelling Discussion">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#structure-your-narrative" class="md-nav__link">
    <span class="md-ellipsis">
      Structure Your Narrative
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-discussion-paragraph" class="md-nav__link">
    <span class="md-ellipsis">
      Example Discussion Paragraph
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#complete-workflow-example" class="md-nav__link">
    <span class="md-ellipsis">
      Complete Workflow Example
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alternative-approaches-and-tools" class="md-nav__link">
    <span class="md-ellipsis">
      Alternative Approaches and Tools
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Alternative Approaches and Tools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-goplot-for-circular-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Using GOplot for Circular Visualization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#revigo-for-semantic-clustering" class="md-nav__link">
    <span class="md-ellipsis">
      REVIGO for Semantic Clustering
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#shinygo-for-interactive-exploration" class="md-nav__link">
    <span class="md-ellipsis">
      ShinyGO for Interactive Exploration
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tips-for-publication-quality-figures" class="md-nav__link">
    <span class="md-ellipsis">
      Tips for Publication-Quality Figures
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#common-pitfalls-to-avoid" class="md-nav__link">
    <span class="md-ellipsis">
      Common Pitfalls to Avoid
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#use-cases-from-our-lab" class="md-nav__link">
    <span class="md-ellipsis">
      Use Cases from Our Lab
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#additional-resources" class="md-nav__link">
    <span class="md-ellipsis">
      Additional Resources
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genome-features" class="md-nav__link">
    <span class="md-ellipsis">
      Genome features
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Genome features">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#bedtoolsintersectbed" class="md-nav__link">
    <span class="md-ellipsis">
      bedtools::intersectbed
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#transcriptome-trinity" class="md-nav__link">
    <span class="md-ellipsis">
      Transcriptome (Trinity)
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/RobertsLab/resources/blob/master/docs/bio-Annotation.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75z"/></svg>
    </a>
  
  


<h1 id="annotation">Annotation</h1>
<h2 id="intro-blast">Intro (Blast)</h2>
<p>Blast is a key component of working with lesser studied taxa. Here are some resources to help with this.</p>
<p><strong>First off, you should be familar with the command line interface and bash</strong></p>
<ul>
<li><a href="https://swcarpentry.github.io/shell-novice/index.html">Introducing the Shell</a></li>
<li><a href="https://datacarpentry.org/shell-genomics/">Introduction to the Command Line for Genomics</a></li>
<li><a href="https://explainshell.com/">https://explainshell.com/</a></li>
</ul>
<hr />
<h3 id="blast-notebooks">Blast Notebooks</h3>
<ul>
<li><a href="https://robertslab.github.io/tusk/modules/04-blast.html">https://robertslab.github.io/tusk/modules/04-blast.html</a> - Blast and annotation on <strong>tusk</strong>.    </li>
<li><a href="https://rpubs.com/sr320/1026094">https://rpubs.com/sr320/1026094</a> - Blast tutorial from FISH 546 (2023)</li>
<li>
<p><a href="https://github.com/RobertsLab/code/blob/master/09-blast.ipynb">https://github.com/RobertsLab/code/blob/master/09-blast.ipynb</a> - An example of how one might take a multi sequence fasta file and using NCBI Blast, compare the sequences with the Swiss-Prot Database on their own computer.</p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/code/blob/master/10-blast-2-slim.ipynb">https://github.com/RobertsLab/code/blob/master/10-blast-2-slim.ipynb</a> - A notebook to seamlessly take blast output to GO Slim list</p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/code/blob/master/script-box/complete_go_annotation_notebook.Rmd">https://github.com/RobertsLab/code/blob/master/script-box/complete_go_annotation_notebook.Rmd</a> -</p>
</li>
<li>
<p>https://github.com/sr320/ceabigr/blob/main/code/17-Swiss-Prot-Annotation.Rmd -  Blasting C virginica to Swiss-Prot. Author: Steven Roberts <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/sr320/ceabigr" /></p>
</li>
</ul>
<hr />
<h2 id="gene-ontology-go">Gene Ontology (GO)</h2>
<h3 id="retrieve-go-terms-from-uniprot-using-swissprot-ids">Retrieve GO terms from UniProt Using SwissProt IDs</h3>
<p>The following steps will use the UniProt Python API to create a tab-delimited file of data retrieved from UniProt.</p>
<ol>
<li>
<p>Create newline-delimited file of SwissProt IDs. (e.g. <code>SPIDS.txt</code>)</p>
<div class="highlight"><pre><span></span><code>cat<span class="w"> </span>SPIDS.txt

Q86IC9
P04177
Q8L840
Q61043
A1E2V0
P34456
P34457
O00463
Q00945
Q5SWK7
</code></pre></div>
</li>
<li>
<p>Create Python file (e.g. <code>uniprot-retrieval.py</code>) with the following:</p>
<div class="highlight"><pre><span></span><code><span class="kn">import</span><span class="w"> </span><span class="nn">re</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">zlib</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">gzip</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">requests</span>
<span class="kn">from</span><span class="w"> </span><span class="nn">requests.adapters</span><span class="w"> </span><span class="kn">import</span> <span class="n">HTTPAdapter</span><span class="p">,</span> <span class="n">Retry</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">sys</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">shutil</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">os</span>

<span class="n">re_next_link</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="sa">r</span><span class="s1">&#39;&lt;(.+)&gt;; rel=&quot;next&quot;&#39;</span><span class="p">)</span>
<span class="n">retries</span> <span class="o">=</span> <span class="n">Retry</span><span class="p">(</span><span class="n">total</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">backoff_factor</span><span class="o">=</span><span class="mf">0.25</span><span class="p">,</span> <span class="n">status_forcelist</span><span class="o">=</span><span class="p">[</span><span class="mi">500</span><span class="p">,</span> <span class="mi">502</span><span class="p">,</span> <span class="mi">503</span><span class="p">,</span> <span class="mi">504</span><span class="p">])</span>
<span class="n">session</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">Session</span><span class="p">()</span>
<span class="n">session</span><span class="o">.</span><span class="n">mount</span><span class="p">(</span><span class="s2">&quot;https://&quot;</span><span class="p">,</span> <span class="n">HTTPAdapter</span><span class="p">(</span><span class="n">max_retries</span><span class="o">=</span><span class="n">retries</span><span class="p">))</span>

<span class="k">def</span><span class="w"> </span><span class="nf">get_next_link</span><span class="p">(</span><span class="n">headers</span><span class="p">):</span>
    <span class="k">if</span> <span class="s2">&quot;Link&quot;</span> <span class="ow">in</span> <span class="n">headers</span><span class="p">:</span>
        <span class="n">match</span> <span class="o">=</span> <span class="n">re_next_link</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">headers</span><span class="p">[</span><span class="s2">&quot;Link&quot;</span><span class="p">])</span>
        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>

<span class="k">def</span><span class="w"> </span><span class="nf">get_batch</span><span class="p">(</span><span class="n">batch_url</span><span class="p">):</span>
    <span class="k">while</span> <span class="n">batch_url</span><span class="p">:</span>
        <span class="n">response</span> <span class="o">=</span> <span class="n">session</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">batch_url</span><span class="p">)</span>
        <span class="n">response</span><span class="o">.</span><span class="n">raise_for_status</span><span class="p">()</span>
        <span class="n">total</span> <span class="o">=</span> <span class="n">response</span><span class="o">.</span><span class="n">headers</span><span class="p">[</span><span class="s2">&quot;x-total-results&quot;</span><span class="p">]</span>
        <span class="k">yield</span> <span class="n">response</span><span class="p">,</span> <span class="n">total</span>
        <span class="n">batch_url</span> <span class="o">=</span> <span class="n">get_next_link</span><span class="p">(</span><span class="n">response</span><span class="o">.</span><span class="n">headers</span><span class="p">)</span>

<span class="k">def</span><span class="w"> </span><span class="nf">process_accessions</span><span class="p">(</span><span class="n">accessions</span><span class="p">):</span>
    <span class="n">accession_batches</span> <span class="o">=</span> <span class="p">[</span><span class="n">accessions</span><span class="p">[</span><span class="n">i</span><span class="p">:</span><span class="n">i</span><span class="o">+</span><span class="mi">500</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">accessions</span><span class="p">),</span> <span class="mi">500</span><span class="p">)]</span>
    <span class="n">all_lines</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">for</span> <span class="n">accession_batch</span> <span class="ow">in</span> <span class="n">accession_batches</span><span class="p">:</span>
        <span class="n">accession_query</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="si">%29%</span><span class="s1">20OR</span><span class="si">%20%</span><span class="s1">28accession%3A&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">accession_batch</span><span class="p">)</span>
        <span class="n">url</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">&quot;https://rest.uniprot.org/uniprotkb/search?compressed=true&amp;fields=accession%2Creviewed%2Cid%2Cprotein_name%2Cgene_names%2Corganism_name%2Clength%2Cgo_p%2Cgo_c%2Cgo%2Cgo_f%2Cgo_id&amp;format=tsv&amp;query=%28%28accession%3A</span><span class="si">{</span><span class="n">accession_query</span><span class="si">}</span><span class="s2">%29%29&amp;size=500&quot;</span>

        <span class="n">progress</span> <span class="o">=</span> <span class="mi">0</span>
        <span class="n">lines</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">batch</span><span class="p">,</span> <span class="n">total</span> <span class="ow">in</span> <span class="n">get_batch</span><span class="p">(</span><span class="n">url</span><span class="p">):</span>
            <span class="c1"># Decompress each batch as we want to extract the header</span>
            <span class="n">decompressed</span> <span class="o">=</span> <span class="n">zlib</span><span class="o">.</span><span class="n">decompress</span><span class="p">(</span><span class="n">batch</span><span class="o">.</span><span class="n">content</span><span class="p">,</span> <span class="mi">16</span> <span class="o">+</span> <span class="n">zlib</span><span class="o">.</span><span class="n">MAX_WBITS</span><span class="p">)</span>
            <span class="n">batch_lines</span> <span class="o">=</span> <span class="p">[</span><span class="n">line</span> <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">decompressed</span><span class="o">.</span><span class="n">decode</span><span class="p">(</span><span class="s2">&quot;utf-8&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span> <span class="k">if</span> <span class="n">line</span><span class="p">]</span>
            <span class="k">if</span> <span class="ow">not</span> <span class="n">progress</span><span class="p">:</span>
                <span class="c1"># First line so print TSV header</span>
                <span class="n">lines</span> <span class="o">=</span> <span class="p">[</span><span class="n">batch_lines</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span>
            <span class="n">lines</span> <span class="o">+=</span> <span class="n">batch_lines</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span>
            <span class="n">progress</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">lines</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1</span>
            <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">progress</span><span class="si">}</span><span class="s2"> / </span><span class="si">{</span><span class="n">total</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>

        <span class="n">all_lines</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="n">lines</span><span class="p">)</span>

    <span class="k">return</span> <span class="n">all_lines</span>

<span class="k">def</span><span class="w"> </span><span class="nf">main</span><span class="p">(</span><span class="n">accession_file</span><span class="p">,</span> <span class="n">output_dir</span><span class="p">):</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">accession_file</span><span class="p">,</span> <span class="s1">&#39;r&#39;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">accessions</span> <span class="o">=</span> <span class="n">f</span><span class="o">.</span><span class="n">read</span><span class="p">()</span><span class="o">.</span><span class="n">splitlines</span><span class="p">()</span>

    <span class="n">retrieved_data</span> <span class="o">=</span> <span class="n">process_accessions</span><span class="p">(</span><span class="n">accessions</span><span class="p">)</span>

    <span class="c1"># Write to a temporary gzip file</span>
    <span class="n">temp_filename</span> <span class="o">=</span> <span class="s2">&quot;uniprot-retrieval-temp.tsv.gz&quot;</span>
    <span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="n">temp_filename</span><span class="p">,</span> <span class="s2">&quot;wt&quot;</span><span class="p">,</span> <span class="n">encoding</span><span class="o">=</span><span class="s2">&quot;utf-8&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="n">f</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">&#39;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">retrieved_data</span><span class="p">))</span>

    <span class="c1"># Determine the full output path</span>
    <span class="n">output_path</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">output_dir</span><span class="p">,</span> <span class="s2">&quot;uniprot-retrieval.tsv.gz&quot;</span><span class="p">)</span>

    <span class="c1"># Merge the temporary file with the existing output file (if it exists)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="n">output_path</span><span class="p">,</span> <span class="s2">&quot;rb&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_existing</span><span class="p">,</span> <span class="nb">open</span><span class="p">(</span><span class="n">temp_filename</span><span class="p">,</span> <span class="s2">&quot;rb&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_temp</span><span class="p">:</span>
            <span class="k">with</span> <span class="n">gzip</span><span class="o">.</span><span class="n">open</span><span class="p">(</span><span class="s2">&quot;uniprot-retrieval-merged.tsv.gz&quot;</span><span class="p">,</span> <span class="s2">&quot;wb&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f_merged</span><span class="p">:</span>
                <span class="n">shutil</span><span class="o">.</span><span class="n">copyfileobj</span><span class="p">(</span><span class="n">f_existing</span><span class="p">,</span> <span class="n">f_merged</span><span class="p">)</span>
                <span class="n">shutil</span><span class="o">.</span><span class="n">copyfileobj</span><span class="p">(</span><span class="n">f_temp</span><span class="p">,</span> <span class="n">f_merged</span><span class="p">)</span>

        <span class="c1"># Replace the original output file with the merged file</span>
        <span class="n">shutil</span><span class="o">.</span><span class="n">move</span><span class="p">(</span><span class="s2">&quot;uniprot-retrieval-merged.tsv.gz&quot;</span><span class="p">,</span> <span class="n">output_path</span><span class="p">)</span>
    <span class="k">except</span> <span class="ne">FileNotFoundError</span><span class="p">:</span>
        <span class="c1"># If the existing output file doesn&#39;t exist, rename the temporary file</span>
        <span class="n">shutil</span><span class="o">.</span><span class="n">move</span><span class="p">(</span><span class="n">temp_filename</span><span class="p">,</span> <span class="n">output_path</span><span class="p">)</span>

<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">3</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Usage: python uniprot-retrieval.py &lt;input_file&gt; &lt;output_directory&gt;&quot;</span><span class="p">)</span>
        <span class="n">sys</span><span class="o">.</span><span class="n">exit</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>

    <span class="n">accession_file</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
    <span class="n">output_dir</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span>

    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">output_dir</span><span class="p">):</span>
        <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">output_dir</span><span class="p">)</span>

    <span class="n">main</span><span class="p">(</span><span class="n">accession_file</span><span class="p">,</span> <span class="n">output_dir</span><span class="p">)</span>
</code></pre></div>
</li>
<li>
<p>Run the Python script:</p>
<p><div class="highlight"><pre><span></span><code>python3<span class="w"> </span>uniprot-retrieval.py<span class="w"> </span>SPIDS.txt<span class="w"> </span>/path/to/desired/output/directory/
</code></pre></div>
- IMPORTANT: Requires Python &gt;= 3!</p>
<ul>
<li>If using R Markdown, run the above in a <code>bash</code> chunk:</li>
</ul>
</li>
</ol>
<p>The resulting output file (<code>uniprot-retrieval.tsv.gz</code>) will be in the specified directory.</p>
<ol>
<li>
<p>Gunzip the output file:</p>
<div class="highlight"><pre><span></span><code>gunzip<span class="w"> </span>uniprot-retrieval.tsv.gz
</code></pre></div>
</li>
</ol>
<p>The resulting file (<code>uniprot-retrieval.tsv</code>) will be formatted with the following columns:</p>
<table>
<thead>
<tr>
<th>Entry</th>
<th>Reviewed</th>
<th>Entry Name</th>
<th>Protein names</th>
<th>Gene Names</th>
<th>Organism</th>
<th>Length</th>
<th>Gene Ontology (biological process)</th>
<th>Gene Ontology (cellular component)</th>
<th>Gene Ontology (molecular function)</th>
<th>Gene Ontology (GO)</th>
<th>Gene Ontology IDs</th>
</tr>
</thead>
<tbody>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>NOTES:</p>
<ul>
<li>
<p>This requires Python &gt;= 3 to run. Simplest way to access Python 3 is via a conda environment.</p>
</li>
<li>
<p>On the first attempt, you'll likely need to install the packages that are being imported at the very beginning of the script.</p>
</li>
<li>
<p><a href="https://github.com/RobertsLab/resources/issues">Create an issue</a> if you need help with any of the above.</p>
</li>
</ul>
<h3 id="map-go-ids-to-goslims">Map GO IDs to GOslims</h3>
<p>Below are a series of R Markdown chunks to run.</p>
<p>The expected input file has at least two columns. One each with:</p>
<ul>
<li>gene ID</li>
<li>Gene Ontology (GO) ID</li>
</ul>
<p>NOTE: The GO IDs in the GO ID column should be separated with a semi-colon.</p>
<p>The basic output from this process will be:</p>
<ul>
<li>GOslim IDs (as rownames)</li>
<li>GOslim terms</li>
<li>Counts of GO IDs matching to corresponding GOslim</li>
<li>Percentage of GO IDs matching to corresponding GOslim</li>
<li>GOIDs mapped to corresponding GOslim, in a semi-colon delimited format</li>
</ul>
<p>There are steps after this that perform different subsetting that you may/not be interested in. They've been left in to serve as examples.</p>
<p>Load libraries</p>
<div class="highlight"><pre><span></span><code>```{r setup, include=TRUE}
library(GSEABase)
library(GO.db)
library(knitr)
library(tidyverse)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = &quot;&quot;         # Prevents appending &#39;##&#39; to beginning of lines in code output
)
```
</code></pre></div>
<p>Variables</p>
<p>IMPORTANT: The user needs to provide:</p>
<ul>
<li>
<p>names of the columns containing the GO IDs and the gene IDs!</p>
</li>
<li>
<p>Path or URL to input file.</p>
</li>
</ul>
<p>After that, there's almost no need to modify any of the chunks which follow.</p>
<div class="highlight"><pre><span></span><code>```{r set-variables, eval=TRUE}
# Column names corresponding to gene name/ID and GO IDs
GO.ID.column &lt;- &quot;Gene.Ontology.IDs&quot;
gene.ID.column &lt;- &quot;gene_id&quot;

# Relative path or URL to input file
input.file &lt;- &quot;https://raw.githubusercontent.com/grace-ac/paper-pycno-sswd-2021-2022/d1cdf13c36085868df4ef4b75d2b7de03ef08d1c/analyses/25-compare-2021-2022/DEGlist_same_2021-2022_forGOslim.tab&quot;


##### Official GO info - no need to change #####
goslims_obo &lt;- &quot;goslim_generic.obo&quot;
goslims_url &lt;- &quot;http://current.geneontology.org/ontology/subsets/goslim_generic.obo&quot;
```
</code></pre></div>
<p>Set GSEAbase location and download <code>goslim_generic.obo</code>
<div class="highlight"><pre><span></span><code>```{r download-generic-goslim-obo, eval=TRUE}
# Find GSEAbase installation location
gseabase_location &lt;- find.package(&quot;GSEABase&quot;)

# Load path to GOslim OBO file
goslim_obo_destintation &lt;- file.path(gseabase_location, &quot;extdata&quot;, goslims_obo, fsep = &quot;/&quot;)

# Download the GOslim OBO file
download.file(url = goslims_url, destfile = goslim_obo_destintation)

# Loads package files
gseabase_files &lt;- system.file(&quot;extdata&quot;, goslims_obo, package=&quot;GSEABase&quot;)
```
</code></pre></div></p>
<p>Read in gene/GO file
<div class="highlight"><pre><span></span><code>```{r read-in-gene-file, eval=TRUE}
full.gene.df &lt;- read.csv(file = input.file, header = TRUE, sep = &quot;\t&quot;)

str(full.gene.df)
```
</code></pre></div></p>
<p>Remove rows with NA, remove whitespace in GO IDs column and keep just gene/GO IDs columns
<div class="highlight"><pre><span></span><code>```{r remove-NA-and-uniprotIDs, eval=TRUE}

# Clean whitespace, filter NA/empty rows, select columns, and split GO terms using column name variables
gene.GO.df &lt;- full.gene.df %&gt;%
  mutate(!!GO.ID.column := str_replace_all(.data[[GO.ID.column]], &quot;\\s*;\\s*&quot;, &quot;;&quot;)) %&gt;% # Clean up spaces around &quot;;&quot;
  filter(!is.na(.data[[gene.ID.column]]) &amp; !is.na(.data[[GO.ID.column]]) &amp; .data[[GO.ID.column]] != &quot;&quot;) %&gt;% 
  select(all_of(c(gene.ID.column, GO.ID.column)))


str(gene.GO.df)
```
</code></pre></div></p>
<p>This flattens the file so all of the GO IDs per gene
are separated into one GO ID per gene per row.
<div class="highlight"><pre><span></span><code>```{r flatten-gene-and-GO-IDs, eval=TRUE}
flat.gene.GO.df &lt;- gene.GO.df %&gt;% separate_rows(!!sym(GO.ID.column), sep = &quot;;&quot;)

str(flat.gene.GO.df)
```
</code></pre></div></p>
<p>Groups the genes by GO ID (i.e. lists all genes associated with each unique GO ID)
<div class="highlight"><pre><span></span><code>```{r group-by-GO, eval=TRUE}
grouped.gene.GO.df &lt;- flat.gene.GO.df %&gt;%
  group_by(!!sym(GO.ID.column)) %&gt;%
  summarise(!!gene.ID.column := paste(.data[[gene.ID.column]], collapse = &quot;,&quot;))

str(grouped.gene.GO.df)
```
</code></pre></div></p>
<p>Map GO IDs to GOslims</p>
<p>The mapping steps were derived from this <a href="https://support.bioconductor.org/p/128407/#128408">bioconductor forum response</a>
<div class="highlight"><pre><span></span><code>```{r vectorize-GOIDs, eval=TRUE}
# Vector of GO IDs
go_ids &lt;- grouped.gene.GO.df[[GO.ID.column]]

str(go_ids)
```
</code></pre></div></p>
<p>Creates new OBO Collection object of just GOslims, based on provided GO IDs.
<div class="highlight"><pre><span></span><code>```{r extract-GOslims-from-OBO, eval=TRUE}

# Create GSEAbase GOCollection using `go_ids`
myCollection &lt;- GOCollection(go_ids)

# Retrieve GOslims from GO OBO file set
slim &lt;- getOBOCollection(gseabase_files)

str(slim)
```
</code></pre></div></p>
<p>Get Biological Process (BP) GOslims associated with provided GO IDs.
<div class="highlight"><pre><span></span><code>```{r retrieve-BP-GOslims, eval=TRUE}
# Retrieve Biological Process (BP) GOslims
slimdf &lt;- goSlim(myCollection, slim, &quot;BP&quot;, verbose)
str(slimdf)
```
</code></pre></div></p>
<p>Performs mapping of of GOIDs to GOslims</p>
<p>Returns:</p>
<ul>
<li>GOslim IDs (as rownames)</li>
<li>GOslim terms</li>
<li>Counts of GO IDs matching to corresponding GOslim</li>
<li>Percentage of GO IDs matching to corresponding GOslim</li>
<li>GOIDs mapped to corresponding GOslim, in a semi-colon delimited format
<div class="highlight"><pre><span></span><code>```{r map-GO-to-GOslims, eval=TRUE}
# List of GOslims and all GO IDs from `go_ids`
gomap &lt;- as.list(GOBPOFFSPRING[rownames(slimdf)])

# Maps `go_ids` to matching GOslims
mapped &lt;- lapply(gomap, intersect, ids(myCollection))

# Append all mapped GO IDs to `slimdf`
# `sapply` needed to apply paste() to create semi-colon delimited values
slimdf$GO.IDs &lt;- sapply(lapply(gomap, intersect, ids(myCollection)), paste, collapse=&quot;;&quot;)

# Remove &quot;character(0) string from &quot;GO.IDs&quot; column
slimdf$GO.IDs[slimdf$GO.IDs == &quot;character(0)&quot;] &lt;- &quot;&quot;

# Add self-matching GOIDs to &quot;GO.IDs&quot; column, if not present
for (go_id in go_ids) {
  # Check if the go_id is present in the row names
  if (go_id %in% rownames(slimdf)) {
    # Check if the go_id is not present in the GO.IDs column
    # Also removes white space &quot;trimws()&quot; and converts all to upper case to handle
    # any weird, &quot;invisible&quot; formatting issues.
    if (!go_id %in% trimws(toupper(strsplit(slimdf[go_id, &quot;GO.IDs&quot;], &quot;;&quot;)[[1]]))) {
      # Append the go_id to the GO.IDs column with a semi-colon separator
      if (length(slimdf$GO.IDs) &gt; 0 &amp;&amp; nchar(slimdf$GO.IDs[nrow(slimdf)]) &gt; 0) {
        slimdf[go_id, &quot;GO.IDs&quot;] &lt;- paste0(slimdf[go_id, &quot;GO.IDs&quot;], &quot;; &quot;, go_id)
      } else {
        slimdf[go_id, &quot;GO.IDs&quot;] &lt;- go_id
      }
    }
  }
}

str(slimdf)
```
</code></pre></div></li>
</ul>
<p>"Flatten" file so each row is single GO ID with corresponding GOslim rownames_to_column needed to retain row name info
<div class="highlight"><pre><span></span><code>```{r flatten-GOslims-file, eval=TRUE}
# &quot;Flatten&quot; file so each row is single GO ID with corresponding GOslim
# rownames_to_column needed to retain row name info
slimdf_separated &lt;- as.data.frame(slimdf %&gt;%
  rownames_to_column(&#39;GOslim&#39;) %&gt;%
  separate_rows(GO.IDs, sep = &quot;;&quot;))

# Group by unique GO ID
grouped_slimdf &lt;- slimdf_separated %&gt;%
  filter(!is.na(GO.IDs) &amp; GO.IDs != &quot;&quot;) %&gt;%
  group_by(GO.IDs) %&gt;%
  summarize(GOslim = paste(GOslim, collapse = &quot;;&quot;),
            Term = paste(Term, collapse = &quot;;&quot;))


str(grouped_slimdf)
```
</code></pre></div></p>
<p>Sorts GOslims by <code>Count</code>, in descending order and then
selects just the <code>Term</code> and <code>Count</code> columns.
<div class="highlight"><pre><span></span><code>```{r sort-and-select-slimdf-counts, eval=TRUE}

slimdf.sorted &lt;- slimdf %&gt;% arrange(desc(Count))

slim.count.df &lt;- slimdf.sorted %&gt;% 
  select(Term, Count)

str(slim.count.df)
```
</code></pre></div></p>
<hr />
<h2 id="enrichment-analysis-visualization-and-interpretation">Enrichment Analysis Visualization and Interpretation</h2>
<p>You've completed your enrichment analysis and have a list of significantly enriched GO terms, KEGG pathways, or other functional categories. Now what? Enrichment analysis often produces hundreds of significantly enriched terms, making it challenging to distill these results into a compelling narrative and impactful visualizations for your research story.</p>
<h3 id="overview-from-results-to-story">Overview: From Results to Story</h3>
<p>After completing enrichment analysis, you typically face these challenges:</p>
<ol>
<li><strong>Identify the most important physiological processes</strong> relevant to your biological question from hundreds of enriched terms</li>
<li><strong>Create visualizations</strong> that clearly communicate key findings without overwhelming readers</li>
<li><strong>Write a coherent discussion</strong> that tells a story rather than listing enriched terms</li>
</ol>
<p>This section provides strategies, code examples, and best practices for transforming your enrichment results into publication-ready figures and narratives. <strong>This guide assumes you already have enrichment results</strong> (e.g., from clusterProfiler, GOseq, DAVID, or similar tools).</p>
<hr />
<h3 id="best-practices-for-synthesizing-enrichment-results">Best Practices for Synthesizing Enrichment Results</h3>
<h4 id="1-prioritize-and-filter-results">1. Prioritize and Filter Results</h4>
<p>Not all significantly enriched terms are equally informative. Consider:</p>
<ul>
<li><strong>Biological relevance</strong>: Focus on processes directly related to your experimental design or biological question</li>
<li><strong>Term specificity</strong>: Prefer specific terms (e.g., "lipid metabolic process") over very broad terms (e.g., "metabolic process")</li>
<li><strong>Statistical significance</strong>: Use adjusted p-values (FDR/q-value &lt; 0.05) and consider effect size (fold enrichment)</li>
<li><strong>Redundancy reduction</strong>: Many GO terms are hierarchical and redundant; simplify by selecting representative terms</li>
</ul>
<p><strong>Strategy</strong>: Start by filtering to the top 10-20 most significant terms, then manually curate to remove redundancy and retain biological meaning.</p>
<h4 id="2-group-related-processes">2. Group Related Processes</h4>
<p>Organize enriched terms into higher-level biological themes:</p>
<ul>
<li><strong>Metabolism</strong>: Energy production, biosynthesis, catabolism</li>
<li><strong>Stress response</strong>: Oxidative stress, heat shock, immune response</li>
<li><strong>Development</strong>: Cell differentiation, morphogenesis, growth</li>
<li><strong>Signaling</strong>: Cell communication, signal transduction</li>
<li><strong>Structural</strong>: Cytoskeleton, cell adhesion, extracellular matrix</li>
</ul>
<p>This grouping helps create a coherent narrative and simplifies visualization.</p>
<h4 id="3-compare-across-conditions">3. Compare Across Conditions</h4>
<p>If you have multiple comparisons (e.g., multiple treatments or time points):</p>
<ul>
<li>Identify <strong>shared</strong> enriched processes (core response)</li>
<li>Identify <strong>unique</strong> enriched processes (condition-specific responses)</li>
<li>Look for <strong>temporal patterns</strong> (early vs. late response)</li>
<li>Consider <strong>opposing processes</strong> (up-regulated vs. down-regulated genes)</li>
</ul>
<hr />
<h3 id="r-packages-for-visualization">R Packages for Visualization</h3>
<h4 id="essential-packages">Essential Packages</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># Install Bioconductor packages (if needed)</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
<span class="w">    </span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">)</span>

<span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;enrichplot&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;GOSemSim&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;DOSE&quot;</span><span class="p">))</span>

<span class="c1"># Install CRAN packages</span>
<span class="nf">install.packages</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;dplyr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;tidyr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;forcats&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;stringr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;RColorBrewer&quot;</span><span class="p">))</span>
</code></pre></div>
<p><strong>Key packages</strong>:
- <strong>enrichplot</strong>: Advanced visualization functions for enrichment results
- <strong>GOSemSim</strong>: Semantic similarity to reduce redundancy
- <strong>ggplot2</strong>: Customizable plotting
- <strong>dplyr/tidyr</strong>: Data manipulation</p>
<hr />
<h3 id="working-with-your-enrichment-results">Working with Your Enrichment Results</h3>
<p>This section assumes you have enrichment results in one of these formats:</p>
<ol>
<li><strong>clusterProfiler object</strong> (e.g., from <code>enrichGO()</code>, <code>enrichKEGG()</code>)</li>
<li><strong>Data frame</strong> with columns for: term ID, description, p-value, adjusted p-value, gene count, etc.</li>
<li><strong>Table from web tools</strong> (DAVID, Enrichr, etc.) exported as CSV/TSV</li>
</ol>
<h4 id="loading-your-results">Loading Your Results</h4>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">enrichplot</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>

<span class="c1"># If you have a clusterProfiler object already:</span>
<span class="c1"># ego_results &lt;- readRDS(&quot;my_enrichment_results.rds&quot;)</span>

<span class="c1"># If you have a table from another tool, read it:</span>
<span class="c1"># enrichment_df &lt;- read.csv(&quot;enrichment_results.csv&quot;)</span>

<span class="c1"># For this guide, we&#39;ll assume you have a clusterProfiler enrichResult object</span>
<span class="c1"># named &#39;ego_results&#39; with your enrichment analysis results</span>
</code></pre></div>
<h4 id="simplify-results-remove-redundancy">Simplify Results (Remove Redundancy)</h4>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">GOSemSim</span><span class="p">)</span>

<span class="c1"># Assuming you have enrichment results in &#39;ego_results&#39;</span>
<span class="c1"># Simplify GO terms based on semantic similarity</span>
<span class="n">ego_simplified</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">simplify</span><span class="p">(</span><span class="n">ego_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                          </span><span class="n">cutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w">      </span><span class="c1"># Similarity threshold (0-1)</span>
<span class="w">                          </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;p.adjust&quot;</span><span class="p">,</span><span class="w">   </span><span class="c1"># Metric to select representative</span>
<span class="w">                          </span><span class="n">select_fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">min</span><span class="p">)</span>

<span class="c1"># Further reduce to top N terms</span>
<span class="n">top_n</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">20</span>
<span class="n">ego_top</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ego_simplified</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="nf">min</span><span class="p">(</span><span class="n">top_n</span><span class="p">,</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">ego_simplified</span><span class="p">)),</span><span class="w"> </span><span class="p">]</span>

<span class="c1"># Convert to dataframe for further manipulation</span>
<span class="n">ego_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">ego_simplified</span><span class="p">)</span>
</code></pre></div>
<p><strong>Note</strong>: If your enrichment results are from a non-clusterProfiler source (e.g., DAVID, Enrichr), you can manually filter redundant terms by:
- Selecting the most specific term from hierarchical groups
- Using <a href="http://revigo.irb.hr/">REVIGO</a> to cluster semantically similar terms
- Prioritizing terms with the highest significance and gene counts</p>
<hr />
<h3 id="creating-impactful-figures">Creating Impactful Figures</h3>
<h4 id="1-dot-plot-most-common-and-effective">1. Dot Plot (Most Common and Effective)</h4>
<p>Dot plots show enriched terms with:
- <strong>X-axis</strong>: Gene ratio or fold enrichment
- <strong>Y-axis</strong>: GO terms
- <strong>Dot size</strong>: Number of genes
- <strong>Dot color</strong>: Significance (p-value or q-value)</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Basic dotplot</span>
<span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span>
<span class="w">        </span><span class="n">font.size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span>
<span class="w">        </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Biological Process Enrichment&quot;</span><span class="p">)</span>

<span class="c1"># Enhanced dotplot with custom colors</span>
<span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span>
<span class="w">        </span><span class="n">font.size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">11</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;Top Enriched Biological Processes in DEGs&quot;</span><span class="p">)</span>
</code></pre></div>
<p><strong>Best for</strong>: Showing top 15-25 enriched terms with statistical significance and gene counts at a glance.</p>
<h4 id="2-bar-plot-clear-and-simple">2. Bar Plot (Clear and Simple)</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># Bar plot showing count or gene ratio</span>
<span class="nf">barplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span>
<span class="w">        </span><span class="n">font.size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;GO Enrichment - Biological Process&quot;</span><span class="p">)</span>

<span class="c1"># Horizontal bar plot with custom aesthetics</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">ego_df</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="p">],</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="n">Count</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p.adjust</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Adjusted\np-value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Gene Count&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Top 15 Enriched GO Terms&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">11</span><span class="p">))</span>
</code></pre></div>
<p><strong>Best for</strong>: Simple, publication-ready figures showing the most enriched processes.</p>
<h4 id="3-networkenrichment-map-shows-relationships">3. Network/Enrichment Map (Shows Relationships)</h4>
<p>Network plots display relationships between enriched terms based on shared genes:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Pairwise term similarity</span>
<span class="n">ego_pairwise</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pairwise_termsim</span><span class="p">(</span><span class="n">ego_simplified</span><span class="p">)</span>

<span class="c1"># Enrichment map (network plot)</span>
<span class="nf">emapplot</span><span class="p">(</span><span class="n">ego_pairwise</span><span class="p">,</span><span class="w"> </span>
<span class="w">         </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">,</span>
<span class="w">         </span><span class="n">cex_label_category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span>
<span class="w">         </span><span class="n">layout</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;nicely&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;Enrichment Map of GO Terms&quot;</span><span class="p">)</span>

<span class="c1"># Alternative: use ggraph for more control</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggraph</span><span class="p">)</span>
<span class="nf">emapplot</span><span class="p">(</span><span class="n">ego_pairwise</span><span class="p">,</span><span class="w"> </span>
<span class="w">         </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">,</span>
<span class="w">         </span><span class="n">cex_label_category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span>
<span class="w">         </span><span class="n">layout</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fr&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1"># Fruchterman-Reingold layout</span>
</code></pre></div>
<p><strong>Best for</strong>: Showing how enriched processes relate to each other and identifying clusters of related functions.</p>
<h4 id="4-upset-plot-for-multiple-gene-lists">4. Upset Plot (For Multiple Gene Lists)</h4>
<p>When comparing enrichment across multiple conditions:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Assuming you have multiple enrichment results</span>
<span class="n">ego_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Condition1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ego_cond1</span><span class="p">,</span>
<span class="w">                </span><span class="n">Condition2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ego_cond2</span><span class="p">,</span>
<span class="w">                </span><span class="n">Condition3</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ego_cond3</span><span class="p">)</span>

<span class="c1"># Upset plot</span>
<span class="nf">upsetplot</span><span class="p">(</span><span class="n">ego_list</span><span class="p">)</span>
</code></pre></div>
<p><strong>Best for</strong>: Comparing enriched terms across multiple experimental conditions or time points.</p>
<h4 id="5-heatmap-of-enriched-terms">5. Heatmap of Enriched Terms</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># Heatplot showing gene-term relationships</span>
<span class="nf">heatplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span>
<span class="w">         </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span>
<span class="w">         </span><span class="n">foldChange</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_fc</span><span class="p">)</span><span class="w">  </span><span class="c1"># Optional: include fold-change values</span>

<span class="c1"># Custom heatmap with ggplot2</span>
<span class="nf">library</span><span class="p">(</span><span class="n">pheatmap</span><span class="p">)</span>

<span class="c1"># Prepare matrix: rows = genes, columns = GO terms</span>
<span class="c1"># (Requires some data wrangling)</span>
</code></pre></div>
<p><strong>Best for</strong>: Detailed view of which genes contribute to which enriched terms.</p>
<h4 id="6-gene-concept-network">6. Gene-Concept Network</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># Show genes associated with top enriched terms</span>
<span class="nf">cnetplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span>
<span class="w">         </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span>
<span class="w">         </span><span class="n">foldChange</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_fc</span><span class="p">,</span><span class="w">  </span><span class="c1"># Optional: show gene expression</span>
<span class="w">         </span><span class="n">circular</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">         </span><span class="n">colorEdge</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># Circular layout</span>
<span class="nf">cnetplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span>
<span class="w">         </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span>
<span class="w">         </span><span class="n">circular</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">         </span><span class="n">colorEdge</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
<p><strong>Best for</strong>: Highlighting specific genes driving enrichment in key processes.</p>
<hr />
<h3 id="advanced-visualization-strategies">Advanced Visualization Strategies</h3>
<h4 id="combining-multiple-plots">Combining Multiple Plots</h4>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>

<span class="c1"># Create multiple plots</span>
<span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_bp</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;Biological Process&quot;</span><span class="p">)</span>
<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_mf</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;Molecular Function&quot;</span><span class="p">)</span>
<span class="n">p3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_cc</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;Cellular Component&quot;</span><span class="p">)</span>

<span class="c1"># Combine into one figure</span>
<span class="n">combined_plot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">p1</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">p2</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">p3</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;combined_GO_enrichment.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">combined_plot</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span><span class="w"> </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">300</span><span class="p">)</span>
</code></pre></div>
<h4 id="custom-ordering-and-grouping">Custom Ordering and Grouping</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># Manually select and order terms by biological theme</span>
<span class="n">selected_terms</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span>
<span class="w">  </span><span class="c1"># Metabolism</span>
<span class="w">  </span><span class="s">&quot;lipid metabolic process&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;carbohydrate metabolic process&quot;</span><span class="p">,</span>

<span class="w">  </span><span class="c1"># Stress response</span>
<span class="w">  </span><span class="s">&quot;response to oxidative stress&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;response to heat&quot;</span><span class="p">,</span>

<span class="w">  </span><span class="c1"># Immune</span>
<span class="w">  </span><span class="s">&quot;innate immune response&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;inflammatory response&quot;</span>
<span class="p">)</span>

<span class="c1"># Filter and plot in custom order</span>
<span class="n">ego_custom</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ego_df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">Description</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">selected_terms</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Description</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">selected_terms</span><span class="p">))</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">ego_custom</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GeneRatio</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Description</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p.adjust</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Key Physiological Processes in Response to Treatment&quot;</span><span class="p">)</span>
</code></pre></div>
<h4 id="adding-annotations-and-context">Adding Annotations and Context</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># Add biological context to plot</span>
<span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;rect&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">xmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.3</span><span class="p">,</span><span class="w"> </span><span class="n">xmax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;yellow&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;text&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.4</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Metabolic\nreprogramming&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">           </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">,</span><span class="w"> </span><span class="n">fontface</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">)</span>
</code></pre></div>
<hr />
<h3 id="writing-a-compelling-discussion">Writing a Compelling Discussion</h3>
<h4 id="structure-your-narrative">Structure Your Narrative</h4>
<ol>
<li><strong>Start with the big picture</strong>: What is the overall biological theme?</li>
<li>
<p><em>"Enrichment analysis revealed a coordinated metabolic shift toward lipid catabolism..."</em></p>
</li>
<li>
<p><strong>Present major themes, not individual terms</strong>: Group related processes</p>
</li>
<li>
<p><em>"Three major physiological responses emerged: (1) stress defense, (2) metabolic remodeling, and (3) immune activation"</em></p>
</li>
<li>
<p><strong>Connect to your hypothesis</strong>: Link enrichment to your research question</p>
</li>
<li>
<p><em>"Consistent with our hypothesis that heat stress triggers energy reallocation..."</em></p>
</li>
<li>
<p><strong>Highlight specific processes</strong>: Dive into 2-3 key enriched processes</p>
</li>
<li>
<p><em>"The enrichment of 'oxidative stress response' (q-value = 0.001, 45 genes) suggests..."</em></p>
</li>
<li>
<p><strong>Integrate with gene expression patterns</strong>: Mention direction (up/down)</p>
</li>
<li>
<p><em>"Up-regulated genes were enriched in immune pathways, while down-regulated genes showed enrichment in growth processes"</em></p>
</li>
<li>
<p><strong>Compare to literature</strong>: Reference similar findings or contrasts</p>
</li>
<li>
<p><em>"This aligns with previous transcriptomic studies in oysters exposed to thermal stress (Smith et al., 2020)"</em></p>
</li>
<li>
<p><strong>Acknowledge limitations and complexity</strong>: Be transparent</p>
</li>
<li><em>"While enrichment analysis provides functional insights, individual gene functions and pathway crosstalk require further investigation"</em></li>
</ol>
<h4 id="example-discussion-paragraph">Example Discussion Paragraph</h4>
<div class="highlight"><pre><span></span><code>Gene ontology enrichment analysis of the 1,247 up-regulated genes revealed significant 
overrepresentation of biological processes related to stress response and metabolic 
remodeling (Figure 3A). Specifically, genes involved in &#39;response to oxidative stress&#39; 
(GO:0006979, q &lt; 0.001, 68 genes), &#39;protein folding&#39; (GO:0006457, q = 0.002, 45 genes), 
and &#39;lipid catabolic process&#39; (GO:0016042, q = 0.003, 52 genes) were highly enriched. 
This pattern suggests a coordinated cellular response to thermal stress characterized by 
both protective mechanisms (heat shock proteins, antioxidant enzymes) and metabolic 
adaptation (shift from anabolic to catabolic processes). In contrast, down-regulated 
genes (n = 892) showed enrichment in &#39;cell division&#39; (GO:0051301, q &lt; 0.001) and 
&#39;protein translation&#39; (GO:0006412, q = 0.002), indicating suppression of energy-intensive 
growth processes during stress. These findings align with the concept of stress-induced 
growth-metabolism tradeoffs previously described in marine invertebrates (Jones et al., 2018).
</code></pre></div>
<hr />
<h3 id="complete-workflow-example">Complete Workflow Example</h3>
<p>This example assumes you already have enrichment results from your analysis.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># ===== Visualization Workflow Starting with Enrichment Results =====</span>

<span class="nf">library</span><span class="p">(</span><span class="n">enrichplot</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">GOSemSim</span><span class="p">)</span>

<span class="c1"># STARTING POINT: You already have enrichment results</span>
<span class="c1"># These could be from clusterProfiler, GOseq, or loaded from saved files</span>
<span class="c1"># For this example, we assume you have two enrichResult objects:</span>
<span class="c1"># - ego_up: enrichment results for up-regulated genes</span>
<span class="c1"># - ego_down: enrichment results for down-regulated genes</span>

<span class="c1"># If you saved your results previously, load them:</span>
<span class="c1"># ego_up &lt;- readRDS(&quot;enrichment_results_up.rds&quot;)</span>
<span class="c1"># ego_down &lt;- readRDS(&quot;enrichment_results_down.rds&quot;)</span>

<span class="c1"># 1. Simplify results to remove redundancy</span>
<span class="n">ego_up_simp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">simplify</span><span class="p">(</span><span class="n">ego_up</span><span class="p">,</span><span class="w"> </span><span class="n">cutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;p.adjust&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">select_fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">min</span><span class="p">)</span>
<span class="n">ego_down_simp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">simplify</span><span class="p">(</span><span class="n">ego_down</span><span class="p">,</span><span class="w"> </span><span class="n">cutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;p.adjust&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">select_fun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">min</span><span class="p">)</span>

<span class="c1"># 2. Create publication-ready figure</span>
<span class="n">p_up</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_up_simp</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Up-regulated Genes&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">)</span>

<span class="n">p_down</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_down_simp</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Down-regulated Genes&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">)</span>

<span class="c1"># Combine plots</span>
<span class="nf">library</span><span class="p">(</span><span class="n">patchwork</span><span class="p">)</span>
<span class="n">combined</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">p_up</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">p_down</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;Figure_GO_enrichment.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">combined</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">300</span><span class="p">)</span>

<span class="c1"># 3. Export results table for supplementary materials</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">ego_up_simp</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;Supplementary_Table_GO_up.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="nf">write.csv</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">ego_down_simp</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;Supplementary_Table_GO_down.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="c1"># 4. Create enrichment map for manuscript</span>
<span class="n">ego_up_pairwise</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pairwise_termsim</span><span class="p">(</span><span class="n">ego_up_simp</span><span class="p">)</span>
<span class="n">p_network</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">emapplot</span><span class="p">(</span><span class="n">ego_up_pairwise</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">ggtitle</span><span class="p">(</span><span class="s">&quot;Enrichment Network - Up-regulated Genes&quot;</span><span class="p">)</span>
<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;Figure_enrichment_network.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">p_network</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">300</span><span class="p">)</span>
</code></pre></div>
<p><strong>Alternative: Working with enrichment results from other tools</strong></p>
<p>If you have enrichment results from DAVID, Enrichr, or similar web tools:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Load your enrichment results table</span>
<span class="n">enrichment_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">&quot;enrichment_results.csv&quot;</span><span class="p">)</span>

<span class="c1"># Assuming columns: Term, P.value, Adjusted.P.value, Genes, Count</span>
<span class="c1"># Filter to significant terms</span>
<span class="n">sig_terms</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">enrichment_df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">Adjusted.P.value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">arrange</span><span class="p">(</span><span class="n">Adjusted.P.value</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">head</span><span class="p">(</span><span class="m">20</span><span class="p">)</span>

<span class="c1"># Create a custom bar plot</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">sig_terms</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">Term</span><span class="p">,</span><span class="w"> </span><span class="n">Count</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Adjusted.P.value</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Adjusted\np-value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Gene Count&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Top 20 Enriched Terms&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">))</span>

<span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;enrichment_barplot.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">300</span><span class="p">)</span>
</code></pre></div>
<hr />
<h3 id="alternative-approaches-and-tools">Alternative Approaches and Tools</h3>
<h4 id="using-goplot-for-circular-visualization">Using GOplot for Circular Visualization</h4>
<div class="highlight"><pre><span></span><code><span class="c1"># Install GOplot</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&quot;GOplot&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
<span class="w">    </span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;GOplot&quot;</span><span class="p">)</span>

<span class="nf">library</span><span class="p">(</span><span class="n">GOplot</span><span class="p">)</span>

<span class="c1"># Prepare data (requires specific format)</span>
<span class="c1"># See: https://wencke.github.io/</span>

<span class="c1"># Circular visualization</span>
<span class="nf">GOCircle</span><span class="p">(</span><span class="n">GOplot_data</span><span class="p">,</span><span class="w"> </span><span class="n">nsub</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span>
</code></pre></div>
<h4 id="revigo-for-semantic-clustering">REVIGO for Semantic Clustering</h4>
<p>For reducing GO term redundancy through semantic similarity:</p>
<ol>
<li>Export GO terms with p-values</li>
<li>Upload to <a href="http://revigo.irb.hr/">REVIGO</a></li>
<li>Visualize clustered terms</li>
<li>Export representative terms</li>
</ol>
<h4 id="shinygo-for-interactive-exploration">ShinyGO for Interactive Exploration</h4>
<p><a href="http://bioinformatics.sdstate.edu/go/">ShinyGO</a> provides a web interface for:
- Enrichment analysis
- Interactive visualizations
- Pathway analysis
- No coding required</p>
<hr />
<h3 id="tips-for-publication-quality-figures">Tips for Publication-Quality Figures</h3>
<ol>
<li><strong>Font sizes</strong>: Ensure labels are readable (minimum 8-10 pt)</li>
<li><strong>Color palettes</strong>: Use colorblind-friendly palettes (viridis, RColorBrewer)</li>
<li><strong>Resolution</strong>: Export at 300 DPI for publication</li>
<li><strong>File format</strong>: PDF for vector graphics, PNG for presentations</li>
<li><strong>Simplicity</strong>: Don't show more than 20-25 terms per figure</li>
<li><strong>Consistency</strong>: Use same color schemes across figures</li>
<li><strong>White space</strong>: Don't overcrowd; consider multi-panel figures</li>
</ol>
<div class="highlight"><pre><span></span><code><span class="c1"># Colorblind-friendly palette</span>
<span class="nf">library</span><span class="p">(</span><span class="n">viridis</span><span class="p">)</span>

<span class="nf">dotplot</span><span class="p">(</span><span class="n">ego_top</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_viridis</span><span class="p">(</span><span class="n">option</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;plasma&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">direction</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">(</span><span class="n">base_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">)</span>
</code></pre></div>
<hr />
<h3 id="common-pitfalls-to-avoid">Common Pitfalls to Avoid</h3>
<ol>
<li><strong>Showing too many terms</strong>: More is not better; focus on top processes</li>
<li><strong>Ignoring redundancy</strong>: Simplify semantically similar terms</li>
<li><strong>Cherry-picking</strong>: Report systematic filtering criteria</li>
<li><strong>Over-interpretation</strong>: GO terms are predictions, not proof of function</li>
<li><strong>Ignoring background</strong>: Always use appropriate universe/background</li>
<li><strong>P-value only</strong>: Consider fold enrichment and gene counts too</li>
<li><strong>No biological context</strong>: Connect enrichment to your research question</li>
</ol>
<hr />
<h3 id="use-cases-from-our-lab">Use Cases from Our Lab</h3>
<ul>
<li>
<p><a href="https://github.com/RobertsLab/paper-tanner-crab">Enrichment analysis examples</a> - Tanner crab transcriptomics <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/RobertsLab/paper-tanner-crab" /></p>
</li>
<li>
<p><a href="https://github.com/sr320/ceabigr">C. virginica GO enrichment</a> - Eastern oyster functional genomics <img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/sr320/ceabigr" /></p>
</li>
</ul>
<hr />
<h3 id="additional-resources">Additional Resources</h3>
<p><strong>R Packages Documentation</strong>:
- <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler book</a>
- <a href="https://github.com/YuLab-SMU/enrichplot">enrichplot documentation</a></p>
<p><strong>Tutorials</strong>:
- <a href="http://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html">clusterProfiler tutorial</a>
- <a href="https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/">GO enrichment analysis workflow</a></p>
<p><strong>Theory</strong>:
- <a href="http://geneontology.org/">Gene Ontology Handbook</a>
- <a href="https://www.pathwaycommons.org/guide/primers/statistics/fishers_exact_test/">Understanding enrichment analysis</a></p>
<hr />
<h2 id="genome-features">Genome features</h2>
<p>In addition to sequence database alignment, finding spatial relationship within a genome is also an import approach for annotation. Often this is done using software tools such as <code>bedtools</code>.</p>
<h3 id="bedtoolsintersectbed"><code>bedtools::intersectbed</code></h3>
<p>see also https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html</p>
<hr />
<h2 id="transcriptome-trinity">Transcriptome (Trinity)</h2>
<p>After <a href="https://robertslab.github.io/resources/bio_Transcriptome-assembly/">transcriptome assembly using Trinity</a>, run the numbered steps below, in order.</p>
<p>NOTE: The following info is long and requires the use of many programs. All of the code listed below are solely examples. Making the commands functional requires a fair amount of organization (i.e. listing paths to programs and input/output files, creating subdirectories for organizing outputs, etc.). See the Use Cases at the end of this section for a more complete picture of how to organize/run this pipeline.</p>
<ol>
<li>
<p>Transdecoder</p>
<ul>
<li>
<p>Identify longest open reading frames (ORFs).</p>
</li>
<li>
<p>Use transcriptome assembly and gene-trans map from <a href="https://robertslab.github.io/resources/bio_Transcriptome-assembly/">Trinity assembly</a>.</p>
<div class="highlight"><pre><span></span><code>TransDecoder.LongOrfs \
--gene_trans_map Trinity.fasta.gene_trans_map \
-t Trinity.fasta&quot;
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>BLASTp</p>
<ul>
<li>
<p>Run blastp on long ORFs from Step 1 above.</p>
</li>
<li>
<p>Output format 6 produces a standard BLAST tab-delimited file.</p>
</li>
<li>
<p>Settings are recommended in TransDecoder documentation.</p>
</li>
<li>
<p>Peptide database (<code>-db uniprot_sprot.pep</code>) is supplied with Trinotate (e.g. <code>Trinotate-v3.1.1/admin/uniprot_sprot.pep</code>), but can be changed to use your own version.</p>
<div class="highlight"><pre><span></span><code>blastp \
-query longest_orfs.pep \
-db uniprot_sprot.pep \
-max_target_seqs 1 \
-outfmt 6 \
-evalue 1e-5 \
-num_threads ${threads} \
&gt; Trinity.fasta.blastp.outfmt6
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>BLASTx (<a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a>)</p>
<ul>
<li>
<p>Run <a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a> blastx on long ORFs from Step 1 above.</p>
</li>
<li>
<p>Output format 6 produces a standard BLAST tab-delimited file.</p>
</li>
<li>
<p>Settings (<code>--evalue</code> and <code>--max-target-seqs</code>) are recommended in TransDecoder documentation.</p>
</li>
<li>
<p><code>--block-size</code> and <code>--index-chunks</code> are specific to running <a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a> BLASTx.</p>
</li>
<li>
<p><code>--db uniprot_sprot.dmnd</code> is a <a href="https://github.com/bbuchfink/diamond"><code>DIAMOND</code></a>-formatted BLAST database. User can generate their own.</p>
<div class="highlight"><pre><span></span><code>```
diamond blastx \
--db uniprot_sprot.dmnd \
--query Trinity.fasta \
--out Trinity.fasta.blastx.outfmt6 \
--outfmt 6 \
--evalue 1e-4 \
--max-target-seqs 1 \
--block-size 15.0 \
--index-chunks 4
```
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>pFam</p>
<ul>
<li>
<p>Run pfam search on long ORFs from Step 1 above.</p>
</li>
<li>
<p>Protein hidden Markov model database (<code>Pfam-A.hmm</code>) is supplied with Trinotate (e.g. <code>Trinotate-v3.1.1/admin/Pfam-A.hmm</code>), but can be changed to use your own version.</p>
<div class="highlight"><pre><span></span><code>hmmscan \
--cpu ${threads} \
--domtblout Trinity.fasta.pfam.domtblout \
Pfam-A.hmm \
longest_orfs.pep
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>Transdecoder</p>
<ul>
<li>Run Transdecoder using transcriptome assembly FastA, <code>blastp</code> and <code>Pfam</code> results.</li>
</ul>
<div class="highlight"><pre><span></span><code>TransDecoder.Predict \
    -t Trinity.fasta \
    --retain_pfam_hits Trinity.fasta.pfam.domtblout \
    --retain_blastp_hits Trinity.fasta.blastp.outfmt6
</code></pre></div>
</li>
<li>
<p>Trinotate</p>
<ul>
<li>
<p>Trinotate requires a large number of steps and programs!</p>
<ul>
<li>
<p>Run <code>signalp</code></p>
<div class="highlight"><pre><span></span><code>signalp \
-f short \
-n Trinity.fasta.trinotate.signalp.out \
longest_orfs.pep
</code></pre></div>
</li>
<li>
<p>Run <code>tmHMM</code></p>
<div class="highlight"><pre><span></span><code>tmhmm \
--short \
&lt; longest_orfs.pep \
&gt; Trinity.fasta.trinotate.tmhmm.out
</code></pre></div>
</li>
<li>
<p>Run <code>RNAmmer</code></p>
<ul>
<li>Uses a special Trinotate implementation of <code>rnammer</code> (e.g. <code>Trinotate/util/rnammer_support/RnammerTranscriptome.pl</code>)</li>
</ul>
<div class="highlight"><pre><span></span><code>RnammerTranscriptome.pl \
--transcriptome Trinity.fasta \
--path_to_rnammer rnammer
</code></pre></div>
</li>
<li>
<p>Load transcripts and coding regions into Trinotate SQLite database</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
init \
--gene_trans_map Trinity.fasta.gene_trans_map \
--transcript_fasta Tinity.fasta \
--transdecoder_pep longest_orfs.pep
</code></pre></div>
</li>
<li>
<p>Load BLASTp/x homologies into SQLite database</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_swissprot_blastp \
Trinity.fasta.blastp.outfmt6
</code></pre></div>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_swissprot_blastx \
Trinity.fasta.blastx.outfmt6
</code></pre></div>
</li>
<li>
<p>Load Pfam into SQLite database</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_pfam \
Trinity.fasta.pfam.domtblout
</code></pre></div>
</li>
<li>
<p>Load transmembrane domains</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_tmhmm \
Trinity.fasta.trinotate.tmhmm.out
</code></pre></div>
</li>
<li>
<p>Load signal peptides</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_signalp \
Trinity.fasta.trinotate.signalp.out
</code></pre></div>
</li>
<li>
<p>Load RNAmmer</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
LOAD_rnammer \
Trinity.fasta.rnammer.gff
</code></pre></div>
</li>
<li>
<p>Create annotation report</p>
<div class="highlight"><pre><span></span><code>Trinotate \
Trinotate.sqlite \
report \
&gt; Trinity.fasta.annotation_report.txt
</code></pre></div>
</li>
<li>
<p>Extract gene ontology (GO) terms from annotation report</p>
<div class="highlight"><pre><span></span><code>extract_GO_assignments_from_Trinotate_xls.pl \
--Trinotate_xls Trinity.fasta.annotation_report.txt \
-G \
--include_ancestral_terms \
&gt; Trinity.fasta.go_annotations.txt
</code></pre></div>
<ul>
<li>The output file is formatted like this (<code>&lt;trinity-id&gt;``&lt;tab&gt;``&lt;GO:NNNNNN,GO:NNNNN,...&gt;</code>):</li>
</ul>
<div class="highlight"><pre><span></span><code>TRINITY_DN0_c0_g1   GO:0003674,GO:0003824,GO:0003964,GO:0006139,GO:0006259,GO:0006310,GO:0006313,GO:0006725,GO:0006807,GO:0008150,GO:0008152,GO:0009987,GO:0016740,GO:0016772,GO:0016779,GO:0032196,GO:0034061,GO:0034641,GO:0043170,GO:0044237,GO:0044238,GO:0044260,GO:0044699,GO:0044710,GO:0044763,GO:0046483,GO:0071704,GO:0090304,GO:1901360
TRINITY_DN0_c10_g1  GO:0003674,GO:0003824,GO:0004659,GO:0004660,GO:0005488,GO:0005575,GO:0005829,GO:0005875,GO:0005965,GO:0006464,GO:0006807,GO:0008150,GO:0008152,GO:0008270,GO:0008318,GO:0009987,GO:0016740,GO:0016765,GO:0018342,GO:0018343,GO:0019538,GO:0032991,GO:0036211,GO:0043167,GO:0043169,GO:0043170,GO:0043234,GO:0043412,GO:0044237,GO:0044238,GO:0044260,GO:0044267,GO:0044422,GO:0044424,GO:0044430,GO:0044444,GO:0044446,GO:0044464,GO:0046872,GO:0046914,GO:0071704,GO:0097354,GO:1901564,GO:1902494,GO:1990234
TRINITY_DN0_c2_g4   GO:0000166,GO:0003674,GO:0005488,GO:0005524,GO:0005575,GO:0005737,GO:0005856,GO:0017076,GO:0030554,GO:0032553,GO:0032555,GO:0032559,GO:0035639,GO:0036094,GO:0043167,GO:0043168,GO:0043226,GO:0043228,GO:0043229,GO:0043232,GO:0044424,GO:0044464,GO:0097159,GO:0097367,GO:1901265,GO:1901363
</code></pre></div>
</li>
</ul>
</li>
<li>
<p>Make transcript features annotation map</p>
<div class="highlight"><pre><span></span><code>```
Trinotate_get_feature_name_encoding_attributes.pl \
Trinity.fasta.annotation_report.txt \
&gt; Trinity.fasta.annotation_feature_map.txt
```
</code></pre></div>
</li>
</ul>
</li>
</ol>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.action.edit", "content.code.copy"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>